{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import matplotlib\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.patches as patches\n",
    "from matplotlib import style, colors\n",
    "from matplotlib import gridspec\n",
    "\n",
    "matplotlib.style.use('ggplot')\n",
    "%matplotlib inline\n",
    "matplotlib.rcParams['figure.figsize'] = (8, 6)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# make oncoprint for 124 patients"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>num_patients_gene_level</th>\n",
       "      <th>num_INDELs_gene_level</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>ref_base</th>\n",
       "      <th>alt_base</th>\n",
       "      <th>num_patients_INDEL_level</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>snp_ID</th>\n",
       "      <th>...</th>\n",
       "      <th>pileup_AltC</th>\n",
       "      <th>pileup_AF</th>\n",
       "      <th>strelka_n_Cov</th>\n",
       "      <th>strelka_n_RefC</th>\n",
       "      <th>strelka_n_AltC</th>\n",
       "      <th>strelka_n_AF</th>\n",
       "      <th>strelka_t_Cov</th>\n",
       "      <th>strelka_t_RefC</th>\n",
       "      <th>strelka_t_AltC</th>\n",
       "      <th>strelka_t_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCA9</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>66992092</td>\n",
       "      <td>C</td>\n",
       "      <td>CA</td>\n",
       "      <td>1</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>novel_snp</td>\n",
       "      <td>...</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>54</td>\n",
       "      <td>51</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>78</td>\n",
       "      <td>41</td>\n",
       "      <td>29</td>\n",
       "      <td>0.41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCB5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>20767995</td>\n",
       "      <td>TATA</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>HTMCP-03-06-02202</td>\n",
       "      <td>novel_snp</td>\n",
       "      <td>...</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>44</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>66</td>\n",
       "      <td>42</td>\n",
       "      <td>18</td>\n",
       "      <td>0.30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene  num_patients_gene_level  num_INDELs_gene_level chromosome  position  \\\n",
       "0  ABCA9                        2                      3         17  66992092   \n",
       "1  ABCB5                        1                      1          7  20767995   \n",
       "\n",
       "  ref_base alt_base  num_patients_INDEL_level         patient_ID     snp_ID  \\\n",
       "0        C       CA                         1  HTMCP-03-06-02007  novel_snp   \n",
       "1     TATA        T                         1  HTMCP-03-06-02202  novel_snp   \n",
       "\n",
       "      ...      pileup_AltC pileup_AF strelka_n_Cov strelka_n_RefC  \\\n",
       "0     ...               na        na            54             51   \n",
       "1     ...               na        na            44             45   \n",
       "\n",
       "  strelka_n_AltC strelka_n_AF strelka_t_Cov strelka_t_RefC strelka_t_AltC  \\\n",
       "0              0          0.0            78             41             29   \n",
       "1              0          0.0            66             42             18   \n",
       "\n",
       "  strelka_t_AF  \n",
       "0         0.41  \n",
       "1         0.30  \n",
       "\n",
       "[2 rows x 29 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f1 ='/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/HIGH_MODERATE_INDEL_summary_with_normal_strelka_only.tsv'\n",
    "indf = pd.read_csv(f1, sep='\\t')\n",
    "indf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>impact_tmp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCA9</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCB5</td>\n",
       "      <td>HTMCP-03-06-02202</td>\n",
       "      <td>inframe_deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABHD14A-ACY1</td>\n",
       "      <td>HTMCP-03-06-02441</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABHD16A</td>\n",
       "      <td>HTMCP-03-06-02435</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AC006035.2</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           gene         patient_ID        impact_tmp\n",
       "0         ABCA9  HTMCP-03-06-02007       FRAME_SHIFT\n",
       "1         ABCB5  HTMCP-03-06-02202  inframe_deletion\n",
       "2  ABHD14A-ACY1  HTMCP-03-06-02441       FRAME_SHIFT\n",
       "3       ABHD16A  HTMCP-03-06-02435       FRAME_SHIFT\n",
       "4    AC006035.2  HTMCP-03-06-02007       FRAME_SHIFT"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indf['impact_tmp'] = indf['snpeff_details'].apply(lambda x: x.split('(')[0])\n",
    "indf = indf[['gene', 'patient_ID','impact_tmp']]\n",
    "indf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['YAP1', 'AJUBA', 'ARID1A', 'ATR', 'ATRX', 'B2M', 'CASP8', 'CBFB', 'ELF3', 'EP300', 'ERBB2', 'ERBB3', 'FAT1', 'FAT2', 'FBXW7', 'HLA-A', 'HLA-B', 'HLA-C', 'KRAS', 'LRP1B', 'MAGEC1', 'MAPK1', 'MLL', 'MLL2', 'MLL3', 'NFE2L2', 'PIK3CA', 'PTEN', 'RB1', 'SF3B1', 'SHKBP1', 'STK11', 'TGFBR2', 'TP53', 'TPR', 'U2AF1', 'ZRSR2']\n"
     ]
    }
   ],
   "source": [
    "# get gene list\n",
    "f2 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/genes.txt'\n",
    "cdf = pd.read_csv(f2)\n",
    "genes = cdf.genes.tolist()\n",
    "print(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>reanne_HIV_status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>reanne_Putative_histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 123 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient                   HTMCP-03-06-02001 HTMCP-03-06-02002  \\\n",
       "reanne_HIV_status                  Positive          Negative   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "\n",
       "patient                   HTMCP-03-06-02003 HTMCP-03-06-02006  \\\n",
       "reanne_HIV_status                  Positive          Negative   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "\n",
       "patient                   HTMCP-03-06-02007 HTMCP-03-06-02008  \\\n",
       "reanne_HIV_status                  Negative          Negative   \n",
       "reanne_Putative_histology             Adeno          Squamous   \n",
       "\n",
       "patient                   HTMCP-03-06-02012 HTMCP-03-06-02013  \\\n",
       "reanne_HIV_status                  Negative          Negative   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "\n",
       "patient                   HTMCP-03-06-02020 HTMCP-03-06-02026  \\\n",
       "reanne_HIV_status                  Positive          Negative   \n",
       "reanne_Putative_histology          Squamous             Adeno   \n",
       "\n",
       "patient                          ...        HTMCP-03-06-02424  \\\n",
       "reanne_HIV_status                ...                 Positive   \n",
       "reanne_Putative_histology        ...                 Squamous   \n",
       "\n",
       "patient                   HTMCP-03-06-02427 HTMCP-03-06-02428  \\\n",
       "reanne_HIV_status                  Positive          Positive   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "\n",
       "patient                   HTMCP-03-06-02434 HTMCP-03-06-02435  \\\n",
       "reanne_HIV_status                  Positive          Positive   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "\n",
       "patient                   HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "reanne_HIV_status                  Positive          Positive   \n",
       "reanne_Putative_histology          Squamous             Adeno   \n",
       "\n",
       "patient                   HTMCP-03-06-02442 HTMCP-03-06-02447  \\\n",
       "reanne_HIV_status                  Positive          Positive   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "\n",
       "patient                   HTMCP-03-06-02448  \n",
       "reanne_HIV_status                  Positive  \n",
       "reanne_Putative_histology             Adeno  \n",
       "\n",
       "[2 rows x 123 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get meta track for example histology\n",
    "f3 ='/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/mutation_load_clinic.txt'\n",
    "ddf = pd.read_csv(f3, sep='\\t')\n",
    "ddf = ddf[['patient', 'reanne_HIV_status', 'reanne_Putative_histology']]\n",
    "edf = ddf.set_index('patient').T\n",
    "edf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/szong/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2717: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "# get snvs\n",
    "f4 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/124_patients_selected_columns_new_filtered.txt'\n",
    "pdf = pd.read_csv(f4, sep='\\t')\n",
    "pdf.head(2)\n",
    "pdf['impact_tmp'] = pdf['snpeff_details'].apply(lambda x: x.split('(')[0])\n",
    "mdf = pdf[['gene', 'patient_ID','impact_tmp']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>impact_tmp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>HTMCP-03-06-02214</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene         patient_ID             impact_tmp\n",
       "0  SAMD11  HTMCP-03-06-02007  NON_SYNONYMOUS_CODING\n",
       "1  SAMD11  HTMCP-03-06-02214  NON_SYNONYMOUS_CODING"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# concat indel and snvs\n",
    "rdf = pd.concat([mdf,indf])\n",
    "rdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>impact_tmp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>HTMCP-03-06-02179</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>HTMCP-03-06-02020</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene         patient_ID             impact_tmp\n",
       "0  A1BG  HTMCP-03-06-02179  NON_SYNONYMOUS_CODING\n",
       "1  A1CF  HTMCP-03-06-02020  NON_SYNONYMOUS_CODING"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf = rdf.drop_duplicates().groupby(['gene', 'patient_ID']).agg({'impact_tmp': ','.join}).reset_index()\n",
    "rdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def impact_type(x):\n",
    "#     print(x)\n",
    "    xsplit = list(set(x.split(',')))\n",
    "    type = xsplit[0].upper()\n",
    "    if len(xsplit) > 1:\n",
    "        impact = 'Multiple'\n",
    "    elif len(xsplit) == 1:\n",
    "         if '+' in xsplit[0]:\n",
    "             impact = 'Multiple'\n",
    "         elif type == 'MISSENSE_VARIANT' or type == 'NON_SYNONYMOUS_CODING':\n",
    "             impact = 'NON_SYNONYMOUS_CODING'\n",
    "         elif  type == 'SPLICE_SITE_ACCEPTOR' or  type == 'SPLICE_SITE_DONOR':\n",
    "             impact = 'SPLICE_ACCEPTOR_DONOR'\n",
    "         elif type == 'STOP_LOST':\n",
    "             impact = 'STOP_LOST'\n",
    "         elif type == 'STOP_GAINED':\n",
    "             impact = 'STOP_GAINED'\n",
    "         elif  type == 'START_LOST':\n",
    "             impact = 'START_LOST'\n",
    "         elif type == 'START_GAINED':\n",
    "             impact = 'START_GAINED'\n",
    "         elif type == 'FRAME_SHIFT' or type == 'FRAMESHIFT_VARIANT':\n",
    "             impact = 'FRAME_SHIFT'\n",
    "         elif type == 'INFRAME_DELETION' or type == 'CODON_INSERTION' or type == 'CODON_DELETION' or type == 'CODON_CHANGE_PLUS_CODON_INSERTION' or type == 'CODON_CHANGE_PLUS_CODON_DELETION' or type.lower() == 'disruptive_inframe_deletion':\n",
    "             impact = 'CODON_INSERTION_DELETION'                \n",
    "         else: exit(1);print('ERROR')\n",
    "    return impact"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient_ID</th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 124 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient_ID HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02003  \\\n",
       "gene                                                               \n",
       "A1BG                    None              None              None   \n",
       "A1CF                    None              None              None   \n",
       "\n",
       "patient_ID HTMCP-03-06-02006 HTMCP-03-06-02007 HTMCP-03-06-02008  \\\n",
       "gene                                                               \n",
       "A1BG                    None              None              None   \n",
       "A1CF                    None              None              None   \n",
       "\n",
       "patient_ID HTMCP-03-06-02012 HTMCP-03-06-02013      HTMCP-03-06-02020  \\\n",
       "gene                                                                    \n",
       "A1BG                    None              None                   None   \n",
       "A1CF                    None              None  NON_SYNONYMOUS_CODING   \n",
       "\n",
       "patient_ID HTMCP-03-06-02026        ...        HTMCP-03-06-02424  \\\n",
       "gene                                ...                            \n",
       "A1BG                    None        ...                     None   \n",
       "A1CF                    None        ...                     None   \n",
       "\n",
       "patient_ID HTMCP-03-06-02427 HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "gene                                                               \n",
       "A1BG                    None              None              None   \n",
       "A1CF                    None              None              None   \n",
       "\n",
       "patient_ID HTMCP-03-06-02435 HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "gene                                                               \n",
       "A1BG                    None              None              None   \n",
       "A1CF                    None              None              None   \n",
       "\n",
       "patient_ID HTMCP-03-06-02442 HTMCP-03-06-02447 HTMCP-03-06-02448  \n",
       "gene                                                              \n",
       "A1BG                    None              None              None  \n",
       "A1CF                    None              None              None  \n",
       "\n",
       "[2 rows x 124 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf['impact'] = rdf['impact_tmp'].apply(lambda x: impact_type(x))\n",
    "rdf = rdf[['gene', 'patient_ID', 'impact']].drop_duplicates()\n",
    "rdf = rdf.set_index(['gene', 'patient_ID'])['impact'].unstack()\n",
    "rdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# of3 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/107_patients/mutations_in_oncoprint.txt'\n",
    "# rdf.fillna('NA').to_csv(of3, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rdf = pd.concat([rdf, edf])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['A1BG', 'A1CF', 'A2LD1', 'A2M', 'A2ML1', 'A4GALT', 'A4GNT', 'AAAS',\n",
       "       'AACS', 'AADAC',\n",
       "       ...\n",
       "       'ZXDC', 'ZYG11A', 'ZYG11B', 'ZZEF1', 'ZZZ3', 'hsa-mir-3187',\n",
       "       'hsa-mir-4763', 'yR211F11.2', 'reanne_HIV_status',\n",
       "       'reanne_Putative_histology'],\n",
       "      dtype='object', length=14575)"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "genes = [ 'reanne_HIV_status', 'reanne_Putative_histology'] + genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Positive', 'Negative', nan, 'Squamous', 'Adeno', None,\n",
       "       'NON_SYNONYMOUS_CODING', 'FRAME_SHIFT', 'Multiple', 'STOP_GAINED',\n",
       "       'SPLICE_ACCEPTOR_DONOR', 'CODON_INSERTION_DELETION'], dtype=object)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf = rdf.reindex(genes)#.loc['PIK3CA'].value_counts()\n",
    "rdf.head(3)\n",
    "pd.Series(rdf.values.ravel()).unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rdf['occurrence'] = (rdf.notnull().sum(axis=1)/(rdf.shape[1] -2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "genes = rdf['occurrence'].sort_values(ascending=False).index.tolist()\n",
    "# genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "      <th>occurrence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>reanne_HIV_status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>1.008197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>reanne_Putative_histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>1.008197</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          HTMCP-03-06-02001 HTMCP-03-06-02002  \\\n",
       "reanne_HIV_status                  Positive          Negative   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "\n",
       "                          HTMCP-03-06-02003 HTMCP-03-06-02006  \\\n",
       "reanne_HIV_status                  Positive          Negative   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "\n",
       "                          HTMCP-03-06-02007 HTMCP-03-06-02008  \\\n",
       "reanne_HIV_status                  Negative          Negative   \n",
       "reanne_Putative_histology             Adeno          Squamous   \n",
       "\n",
       "                          HTMCP-03-06-02012 HTMCP-03-06-02013  \\\n",
       "reanne_HIV_status                  Negative          Negative   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "\n",
       "                          HTMCP-03-06-02020 HTMCP-03-06-02026    ...      \\\n",
       "reanne_HIV_status                  Positive          Negative    ...       \n",
       "reanne_Putative_histology          Squamous             Adeno    ...       \n",
       "\n",
       "                          HTMCP-03-06-02427 HTMCP-03-06-02428  \\\n",
       "reanne_HIV_status                  Positive          Positive   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "\n",
       "                          HTMCP-03-06-02434 HTMCP-03-06-02435  \\\n",
       "reanne_HIV_status                  Positive          Positive   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "\n",
       "                          HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "reanne_HIV_status                  Positive          Positive   \n",
       "reanne_Putative_histology          Squamous             Adeno   \n",
       "\n",
       "                          HTMCP-03-06-02442 HTMCP-03-06-02447  \\\n",
       "reanne_HIV_status                  Positive          Positive   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "\n",
       "                          HTMCP-03-06-02448 occurrence  \n",
       "reanne_HIV_status                  Positive   1.008197  \n",
       "reanne_Putative_histology             Adeno   1.008197  \n",
       "\n",
       "[2 rows x 125 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf = rdf.reindex(genes)\n",
    "rdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rdf['percentage'] = ['{0}({1}%)'.format(i[0], int(round(i[1]*100))) for i in zip(rdf.index.tolist(), rdf.occurrence)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rdf.loc['reanne_HIV_status', 'percentage'] = 'HIV_Status'\n",
    "rdf.loc['reanne_Putative_histology', 'percentage' ] = 'Puatative_Histology'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "      <th>occurrence</th>\n",
       "      <th>percentage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>reanne_HIV_status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>1.008197</td>\n",
       "      <td>HIV_Status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>reanne_Putative_histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>1.008197</td>\n",
       "      <td>Puatative_Histology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CA</th>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>None</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>...</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>0.360656</td>\n",
       "      <td>PIK3CA(36%)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 126 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               HTMCP-03-06-02001      HTMCP-03-06-02002  \\\n",
       "reanne_HIV_status                       Positive               Negative   \n",
       "reanne_Putative_histology               Squamous               Squamous   \n",
       "PIK3CA                     NON_SYNONYMOUS_CODING  NON_SYNONYMOUS_CODING   \n",
       "\n",
       "                          HTMCP-03-06-02003 HTMCP-03-06-02006  \\\n",
       "reanne_HIV_status                  Positive          Negative   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "PIK3CA                                 None              None   \n",
       "\n",
       "                          HTMCP-03-06-02007 HTMCP-03-06-02008  \\\n",
       "reanne_HIV_status                  Negative          Negative   \n",
       "reanne_Putative_histology             Adeno          Squamous   \n",
       "PIK3CA                             Multiple              None   \n",
       "\n",
       "                               HTMCP-03-06-02012      HTMCP-03-06-02013  \\\n",
       "reanne_HIV_status                       Negative               Negative   \n",
       "reanne_Putative_histology               Squamous               Squamous   \n",
       "PIK3CA                     NON_SYNONYMOUS_CODING  NON_SYNONYMOUS_CODING   \n",
       "\n",
       "                          HTMCP-03-06-02020 HTMCP-03-06-02026  \\\n",
       "reanne_HIV_status                  Positive          Negative   \n",
       "reanne_Putative_histology          Squamous             Adeno   \n",
       "PIK3CA                                 None       STOP_GAINED   \n",
       "\n",
       "                                  ...               HTMCP-03-06-02428  \\\n",
       "reanne_HIV_status                 ...                        Positive   \n",
       "reanne_Putative_histology         ...                        Squamous   \n",
       "PIK3CA                            ...           NON_SYNONYMOUS_CODING   \n",
       "\n",
       "                          HTMCP-03-06-02434 HTMCP-03-06-02435  \\\n",
       "reanne_HIV_status                  Positive          Positive   \n",
       "reanne_Putative_histology          Squamous          Squamous   \n",
       "PIK3CA                                 None              None   \n",
       "\n",
       "                          HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "reanne_HIV_status                  Positive          Positive   \n",
       "reanne_Putative_histology          Squamous             Adeno   \n",
       "PIK3CA                                 None              None   \n",
       "\n",
       "                          HTMCP-03-06-02442      HTMCP-03-06-02447  \\\n",
       "reanne_HIV_status                  Positive               Positive   \n",
       "reanne_Putative_histology          Squamous               Squamous   \n",
       "PIK3CA                                 None  NON_SYNONYMOUS_CODING   \n",
       "\n",
       "                          HTMCP-03-06-02448 occurrence           percentage  \n",
       "reanne_HIV_status                  Positive   1.008197           HIV_Status  \n",
       "reanne_Putative_histology             Adeno   1.008197  Puatative_Histology  \n",
       "PIK3CA                                 None   0.360656          PIK3CA(36%)  \n",
       "\n",
       "[3 rows x 126 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sdf = rdf.drop('occurrence', axis=1).set_index('percentage', drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# sdf = sdf.fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>percentage</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_Status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Puatative_Histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CA(36%)</th>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>None</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 124 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         HTMCP-03-06-02001      HTMCP-03-06-02002  \\\n",
       "percentage                                                          \n",
       "HIV_Status                        Positive               Negative   \n",
       "Puatative_Histology               Squamous               Squamous   \n",
       "PIK3CA(36%)          NON_SYNONYMOUS_CODING  NON_SYNONYMOUS_CODING   \n",
       "\n",
       "                    HTMCP-03-06-02003 HTMCP-03-06-02006 HTMCP-03-06-02007  \\\n",
       "percentage                                                                  \n",
       "HIV_Status                   Positive          Negative          Negative   \n",
       "Puatative_Histology          Squamous          Squamous             Adeno   \n",
       "PIK3CA(36%)                      None              None          Multiple   \n",
       "\n",
       "                    HTMCP-03-06-02008      HTMCP-03-06-02012  \\\n",
       "percentage                                                     \n",
       "HIV_Status                   Negative               Negative   \n",
       "Puatative_Histology          Squamous               Squamous   \n",
       "PIK3CA(36%)                      None  NON_SYNONYMOUS_CODING   \n",
       "\n",
       "                         HTMCP-03-06-02013 HTMCP-03-06-02020  \\\n",
       "percentage                                                     \n",
       "HIV_Status                        Negative          Positive   \n",
       "Puatative_Histology               Squamous          Squamous   \n",
       "PIK3CA(36%)          NON_SYNONYMOUS_CODING              None   \n",
       "\n",
       "                    HTMCP-03-06-02026        ...        HTMCP-03-06-02424  \\\n",
       "percentage                                   ...                            \n",
       "HIV_Status                   Negative        ...                 Positive   \n",
       "Puatative_Histology             Adeno        ...                 Squamous   \n",
       "PIK3CA(36%)               STOP_GAINED        ...                     None   \n",
       "\n",
       "                    HTMCP-03-06-02427      HTMCP-03-06-02428  \\\n",
       "percentage                                                     \n",
       "HIV_Status                   Positive               Positive   \n",
       "Puatative_Histology          Squamous               Squamous   \n",
       "PIK3CA(36%)                      None  NON_SYNONYMOUS_CODING   \n",
       "\n",
       "                    HTMCP-03-06-02434 HTMCP-03-06-02435 HTMCP-03-06-02437  \\\n",
       "percentage                                                                  \n",
       "HIV_Status                   Positive          Positive          Positive   \n",
       "Puatative_Histology          Squamous          Squamous          Squamous   \n",
       "PIK3CA(36%)                      None              None              None   \n",
       "\n",
       "                    HTMCP-03-06-02441 HTMCP-03-06-02442  \\\n",
       "percentage                                                \n",
       "HIV_Status                   Positive          Positive   \n",
       "Puatative_Histology             Adeno          Squamous   \n",
       "PIK3CA(36%)                      None              None   \n",
       "\n",
       "                         HTMCP-03-06-02447 HTMCP-03-06-02448  \n",
       "percentage                                                    \n",
       "HIV_Status                        Positive          Positive  \n",
       "Puatative_Histology               Squamous             Adeno  \n",
       "PIK3CA(36%)          NON_SYNONYMOUS_CODING              None  \n",
       "\n",
       "[3 rows x 124 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdf.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sdf.to_csv('/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/mutations_for_oncoprint_124_patients.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
