{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.patches as patches\n",
    "from matplotlib import style, colors\n",
    "from matplotlib import gridspec\n",
    "\n",
    "matplotlib.style.use('ggplot')\n",
    "%matplotlib inline\n",
    "matplotlib.rcParams['figure.figsize'] = (8, 6)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# make oncoprint for 124 patients"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['MAPK1', 'CASP8', 'FBXW7', 'MLL2', 'ZC3H6', 'PCDHGA12', 'FAT1', 'ZNF750', 'SLC35G5', 'PCDHA9', 'PSPC1']\n"
     ]
    }
   ],
   "source": [
    "# get gene list\n",
    "# f2 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/genes_grouped_for_oncoprint.txt'\n",
    "# f2 = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/histogenes_genes_grouped_for_oncoprint.txt'\n",
    "f2 = '/projects/trans_scratch/validations/workspace/szong/Cervical/mutsig2cv/118_patients/smgs_reviewed.txt'\n",
    "cdf = pd.read_csv(f2, comment=\"#\")\n",
    "cdf.columns = ['genes']\n",
    "genes = cdf.genes.tolist()\n",
    "print(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>num_patients_gene_level</th>\n",
       "      <th>num_INDELs_gene_level</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>ref_base</th>\n",
       "      <th>alt_base</th>\n",
       "      <th>num_patients_INDEL_level</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>snp_ID</th>\n",
       "      <th>...</th>\n",
       "      <th>pileup_AltC</th>\n",
       "      <th>pileup_AF</th>\n",
       "      <th>strelka_n_Cov</th>\n",
       "      <th>strelka_n_RefC</th>\n",
       "      <th>strelka_n_AltC</th>\n",
       "      <th>strelka_n_AF</th>\n",
       "      <th>strelka_t_Cov</th>\n",
       "      <th>strelka_t_RefC</th>\n",
       "      <th>strelka_t_AltC</th>\n",
       "      <th>strelka_t_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PCNT</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>47849950</td>\n",
       "      <td>CAGG</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>HTMCP-03-06-02448</td>\n",
       "      <td>novel_snp</td>\n",
       "      <td>...</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>41</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>83</td>\n",
       "      <td>62</td>\n",
       "      <td>15</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MLL3</td>\n",
       "      <td>29</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>151879476</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>HTMCP-03-06-02097</td>\n",
       "      <td>novel_snp</td>\n",
       "      <td>...</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>33</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>84</td>\n",
       "      <td>56</td>\n",
       "      <td>22</td>\n",
       "      <td>0.11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene  num_patients_gene_level  num_INDELs_gene_level chromosome   position  \\\n",
       "0  PCNT                        1                      2         21   47849950   \n",
       "1  MLL3                       29                      6          7  151879476   \n",
       "\n",
       "  ref_base alt_base  num_patients_INDEL_level         patient_ID     snp_ID  \\\n",
       "0     CAGG        C                         1  HTMCP-03-06-02448  novel_snp   \n",
       "1      CTG        C                         1  HTMCP-03-06-02097  novel_snp   \n",
       "\n",
       "      ...      pileup_AltC pileup_AF strelka_n_Cov strelka_n_RefC  \\\n",
       "0     ...               na        na            41             43   \n",
       "1     ...               na        na            33             32   \n",
       "\n",
       "  strelka_n_AltC strelka_n_AF strelka_t_Cov strelka_t_RefC strelka_t_AltC  \\\n",
       "0              0          0.0            83             62             15   \n",
       "1              0          0.0            84             56             22   \n",
       "\n",
       "  strelka_t_AF  \n",
       "0         0.05  \n",
       "1         0.11  \n",
       "\n",
       "[2 rows x 29 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f1 ='/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/HIGH_MODERATE_INDEL_summary_with_normal_strelka_indels_only.tsv'\n",
    "indf = pd.read_csv(f1, sep='\\t')\n",
    "indf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(827, 3)"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indf['impact_tmp'] = indf['snpeff_details'].apply(lambda x: x.split('(')[0])\n",
    "indf = indf[['gene', 'patient_ID','impact_tmp']]\n",
    "indf.head()\n",
    "indf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "indf.drop_duplicates(keep='first', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>impact_tmp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>322</th>\n",
       "      <td>IL2RG</td>\n",
       "      <td>HTMCP-03-06-02157</td>\n",
       "      <td>frameshift_variant+splice_donor_variant+splice...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene         patient_ID  \\\n",
       "322  IL2RG  HTMCP-03-06-02157   \n",
       "\n",
       "                                            impact_tmp  \n",
       "322  frameshift_variant+splice_donor_variant+splice...  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indf[indf.impact_tmp == 'frameshift_variant+stop_gained']\n",
    "indf[indf.impact_tmp == 'frameshift_variant+splice_donor_variant+splice_region_variant+splice_region_variant+intron_variant']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CODON_CHANGE_PLUS_CODON_DELETION\n",
      "CODON_CHANGE_PLUS_CODON_INSERTION\n",
      "CODON_DELETION\n",
      "CODON_INSERTION\n",
      "FRAME_SHIFT\n",
      "SPLICE_SITE_ACCEPTOR\n",
      "SPLICE_SITE_DONOR\n",
      "STOP_GAINED\n",
      "disruptive_inframe_deletion\n",
      "frameshift_variant\n",
      "frameshift_variant+splice_donor_variant+splice_region_variant+splice_region_variant+intron_variant\n",
      "frameshift_variant+stop_gained\n",
      "inframe_deletion\n",
      "splice_donor_variant+disruptive_inframe_deletion+splice_region_variant+intron_variant\n",
      "splice_donor_variant+intron_variant\n",
      "splice_donor_variant+splice_region_variant+splice_region_variant+intron_variant+non_coding_exon_variant\n"
     ]
    }
   ],
   "source": [
    "for i in sorted(indf.impact_tmp.unique()):\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adenosquamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 123 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient            HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02003  \\\n",
       "HIV_status                  Positive          Negative          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02006 HTMCP-03-06-02007 HTMCP-03-06-02008  \\\n",
       "HIV_status                  Negative          Negative          Negative   \n",
       "Putative_histology          Squamous     Adenosquamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02012 HTMCP-03-06-02013 HTMCP-03-06-02020  \\\n",
       "HIV_status                  Negative          Negative          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02026        ...        HTMCP-03-06-02424  \\\n",
       "HIV_status                  Negative        ...                 Positive   \n",
       "Putative_histology             Adeno        ...                 Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02427 HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "HIV_status                  Positive          Positive          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02435 HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "HIV_status                  Positive          Positive          Positive   \n",
       "Putative_histology          Squamous          Squamous             Adeno   \n",
       "\n",
       "patient            HTMCP-03-06-02442 HTMCP-03-06-02447 HTMCP-03-06-02448  \n",
       "HIV_status                  Positive          Positive          Positive  \n",
       "Putative_histology          Squamous          Squamous             Adeno  \n",
       "\n",
       "[2 rows x 123 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get meta track for example histology\n",
    "f3 ='/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/mutation_load_clinic.txt'\n",
    "ddf = pd.read_csv(f3, sep='\\t')\n",
    "ddf = ddf[['patient', 'HIV_status', 'Putative_histology']]\n",
    "edf = ddf.set_index('patient').T\n",
    "edf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Squamous', 'Adenosquamous', 'Adeno', 'Neuroendocrine'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edf.loc['Putative_histology',].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "# get snvs\n",
    "# f4 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/124_patients_selected_columns_new_filtered.txt'\n",
    "f4 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/HIGH_MODERATE_SNV_summary_with_normal_with_AF_filtered_somatic.tsv'\n",
    "pdf = pd.read_csv(f4, sep='\\t')\n",
    "pdf.head(2)\n",
    "pdf['impact_tmp'] = pdf['snpeff_details'].apply(lambda x: x.split('(')[0])\n",
    "mdf = pdf[['gene', 'patient_ID','impact_tmp']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>impact_tmp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>HTMCP-03-06-02214</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene         patient_ID             impact_tmp\n",
       "0  SAMD11  HTMCP-03-06-02007  NON_SYNONYMOUS_CODING\n",
       "1  SAMD11  HTMCP-03-06-02214  NON_SYNONYMOUS_CODING"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>impact_tmp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>HTMCP-03-06-02214</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene         patient_ID             impact_tmp\n",
       "0  SAMD11  HTMCP-03-06-02007  NON_SYNONYMOUS_CODING\n",
       "1  SAMD11  HTMCP-03-06-02214  NON_SYNONYMOUS_CODING"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# concat indel and snvs\n",
    "rdf = pd.concat([mdf,indf])\n",
    "rdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "rdf = rdf[rdf['patient_ID']!=\"HTMCP-03-06-02061\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>impact_tmp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>HTMCP-03-06-02179</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>HTMCP-03-06-02020</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene         patient_ID             impact_tmp\n",
       "0  A1BG  HTMCP-03-06-02179  NON_SYNONYMOUS_CODING\n",
       "1  A1CF  HTMCP-03-06-02020  NON_SYNONYMOUS_CODING"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf = rdf.drop_duplicates().groupby(['gene', 'patient_ID']).agg({'impact_tmp': ','.join}).reset_index()\n",
    "rdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['NON_SYNONYMOUS_CODING', 'STOP_GAINED',\n",
       "       'STOP_GAINED,NON_SYNONYMOUS_CODING', 'SPLICE_SITE_ACCEPTOR',\n",
       "       'NON_SYNONYMOUS_CODING,STOP_GAINED',\n",
       "       'NON_SYNONYMOUS_CODING,STOP_GAINED,SPLICE_SITE_ACCEPTOR',\n",
       "       'missense_variant', 'NON_SYNONYMOUS_CODING,SPLICE_SITE_DONOR',\n",
       "       'NON_SYNONYMOUS_CODING,FRAME_SHIFT',\n",
       "       'SPLICE_SITE_ACCEPTOR,NON_SYNONYMOUS_CODING,STOP_GAINED',\n",
       "       'inframe_deletion', 'SPLICE_SITE_DONOR', 'FRAME_SHIFT',\n",
       "       'NON_SYNONYMOUS_CODING,SPLICE_SITE_ACCEPTOR', 'STOP_LOST',\n",
       "       'NON_SYNONYMOUS_CODING,CODON_CHANGE_PLUS_CODON_DELETION',\n",
       "       'splice_acceptor_variant+intron_variant', 'frameshift_variant',\n",
       "       'START_LOST', 'SPLICE_SITE_ACCEPTOR,NON_SYNONYMOUS_CODING',\n",
       "       'START_LOST,NON_SYNONYMOUS_CODING',\n",
       "       'SPLICE_SITE_DONOR,NON_SYNONYMOUS_CODING',\n",
       "       'NON_SYNONYMOUS_CODING,STOP_GAINED,FRAME_SHIFT',\n",
       "       'CODON_CHANGE_PLUS_CODON_DELETION', 'CODON_DELETION',\n",
       "       'STOP_GAINED,NON_SYNONYMOUS_CODING,SPLICE_SITE_ACCEPTOR,FRAME_SHIFT',\n",
       "       'STOP_GAINED,SPLICE_SITE_ACCEPTOR',\n",
       "       'STOP_GAINED,SPLICE_SITE_ACCEPTOR,NON_SYNONYMOUS_CODING',\n",
       "       'stop_gained', 'CODON_CHANGE_PLUS_CODON_INSERTION',\n",
       "       'SPLICE_SITE_DONOR,SPLICE_SITE_ACCEPTOR',\n",
       "       'NON_SYNONYMOUS_CODING,CODON_DELETION',\n",
       "       'NON_SYNONYMOUS_CODING,SPLICE_SITE_ACCEPTOR,STOP_GAINED',\n",
       "       'missense_variant,NON_SYNONYMOUS_CODING',\n",
       "       'missense_variant+splice_region_variant,missense_variant',\n",
       "       'STOP_GAINED,SPLICE_SITE_DONOR,NON_SYNONYMOUS_CODING',\n",
       "       'NON_SYNONYMOUS_CODING,STOP_LOST',\n",
       "       'NON_SYNONYMOUS_CODING,stop_gained',\n",
       "       'splice_donor_variant+disruptive_inframe_deletion+splice_region_variant+intron_variant',\n",
       "       'start_lost', 'disruptive_inframe_deletion',\n",
       "       'STOP_GAINED,NON_SYNONYMOUS_CODING,SPLICE_SITE_ACCEPTOR',\n",
       "       'STOP_GAINED,NON_SYNONYMOUS_CODING,FRAME_SHIFT',\n",
       "       'SPLICE_SITE_ACCEPTOR,STOP_GAINED',\n",
       "       'NON_SYNONYMOUS_CODING,SPLICE_SITE_DONOR,STOP_GAINED',\n",
       "       'STOP_LOST,NON_SYNONYMOUS_CODING,STOP_GAINED',\n",
       "       'SPLICE_SITE_ACCEPTOR,STOP_GAINED,NON_SYNONYMOUS_CODING',\n",
       "       'STOP_LOST,NON_SYNONYMOUS_CODING',\n",
       "       'NON_SYNONYMOUS_CODING,SPLICE_SITE_DONOR,SPLICE_SITE_ACCEPTOR',\n",
       "       'STOP_GAINED,NON_SYNONYMOUS_CODING,SPLICE_SITE_DONOR',\n",
       "       'FRAME_SHIFT,CODON_DELETION', 'NON_SYNONYMOUS_CODING,START_LOST',\n",
       "       'STOP_GAINED,SPLICE_SITE_DONOR',\n",
       "       'NON_SYNONYMOUS_CODING,STOP_GAINED,SPLICE_SITE_DONOR',\n",
       "       'NON_SYNONYMOUS_CODING,CODON_CHANGE_PLUS_CODON_INSERTION',\n",
       "       'SPLICE_SITE_DONOR,STOP_GAINED',\n",
       "       'STOP_GAINED,NON_SYNONYMOUS_CODING,STOP_LOST',\n",
       "       'NON_SYNONYMOUS_CODING,frameshift_variant+stop_gained',\n",
       "       'SPLICE_SITE_DONOR,NON_SYNONYMOUS_CODING,STOP_GAINED',\n",
       "       'NON_SYNONYMOUS_CODING,STOP_GAINED,STOP_LOST',\n",
       "       'frameshift_variant+splice_donor_variant+splice_region_variant+splice_region_variant+intron_variant',\n",
       "       'stop_lost', 'CODON_INSERTION', 'stop_gained,missense_variant',\n",
       "       'missense_variant+splice_region_variant',\n",
       "       'SPLICE_SITE_DONOR,NON_SYNONYMOUS_CODING,FRAME_SHIFT',\n",
       "       'SPLICE_SITE_DONOR,NON_SYNONYMOUS_CODING,STOP_GAINED,SPLICE_SITE_ACCEPTOR',\n",
       "       'NON_SYNONYMOUS_CODING,STOP_GAINED,CODON_DELETION',\n",
       "       'STOP_GAINED,FRAME_SHIFT',\n",
       "       'SPLICE_SITE_ACCEPTOR,NON_SYNONYMOUS_CODING,STOP_GAINED,FRAME_SHIFT',\n",
       "       'CODON_DELETION,FRAME_SHIFT',\n",
       "       'NON_SYNONYMOUS_CODING,missense_variant',\n",
       "       'missense_variant,STOP_GAINED', 'STOP_GAINED,CODON_DELETION',\n",
       "       'NON_SYNONYMOUS_CODING,SPLICE_SITE_ACCEPTOR,FRAME_SHIFT',\n",
       "       'STOP_GAINED,frameshift_variant',\n",
       "       'splice_donor_variant+intron_variant',\n",
       "       'splice_donor_variant+splice_region_variant+splice_region_variant+intron_variant+non_coding_exon_variant',\n",
       "       'NON_SYNONYMOUS_CODING,missense_variant,stop_gained',\n",
       "       'STOP_LOST,STOP_GAINED,NON_SYNONYMOUS_CODING',\n",
       "       'NON_SYNONYMOUS_CODING,STOP_GAINED,START_LOST',\n",
       "       'NON_SYNONYMOUS_CODING,frameshift_variant',\n",
       "       'STOP_GAINED,NON_SYNONYMOUS_CODING,missense_variant',\n",
       "       'stop_gained,missense_variant,NON_SYNONYMOUS_CODING',\n",
       "       'SPLICE_SITE_DONOR,FRAME_SHIFT'], dtype=object)"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf.impact_tmp.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "def impact_type(x):\n",
    "#     print(x)\n",
    "    xsplit = list(set(x.split(',')))\n",
    "    type = xsplit[0].upper()\n",
    "    if len(xsplit) > 1:\n",
    "        impact = 'Multiple'\n",
    "    elif len(xsplit) == 1:\n",
    "         if '+' in xsplit[0]:\n",
    "             impact = 'Multiple'\n",
    "         elif type == 'MISSENSE_VARIANT' or type == 'NON_SYNONYMOUS_CODING':\n",
    "             impact = 'NON_SYNONYMOUS_CODING'\n",
    "         elif  type == 'SPLICE_SITE_ACCEPTOR' or  type == 'SPLICE_SITE_DONOR':\n",
    "             impact = 'SPLICE_ACCEPTOR_DONOR'\n",
    "         elif type == 'STOP_LOST':\n",
    "             impact = 'STOP_LOST'\n",
    "         elif type == 'STOP_GAINED':\n",
    "             impact = 'STOP_GAINED'\n",
    "         elif  type == 'START_LOST':\n",
    "             impact = 'START_LOST'\n",
    "         elif type == 'START_GAINED':\n",
    "             impact = 'START_GAINED'\n",
    "         elif type == 'FRAME_SHIFT' or type == 'FRAMESHIFT_VARIANT':\n",
    "             impact = 'FRAME_SHIFT'\n",
    "         elif type == 'INFRAME_DELETION' or type == 'CODON_INSERTION' or type == 'CODON_DELETION' or type == 'CODON_CHANGE_PLUS_CODON_INSERTION' or type == 'CODON_CHANGE_PLUS_CODON_DELETION' or type.lower() == 'disruptive_inframe_deletion':\n",
    "             impact = 'CODON_INSERTION_DELETION'                \n",
    "         else: exit(1);print('ERROR')\n",
    "    return impact"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient_ID</th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 123 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient_ID HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02003  \\\n",
       "gene                                                               \n",
       "A1BG                     NaN               NaN               NaN   \n",
       "A1CF                     NaN               NaN               NaN   \n",
       "\n",
       "patient_ID HTMCP-03-06-02006 HTMCP-03-06-02007 HTMCP-03-06-02008  \\\n",
       "gene                                                               \n",
       "A1BG                     NaN               NaN               NaN   \n",
       "A1CF                     NaN               NaN               NaN   \n",
       "\n",
       "patient_ID HTMCP-03-06-02012 HTMCP-03-06-02013      HTMCP-03-06-02020  \\\n",
       "gene                                                                    \n",
       "A1BG                     NaN               NaN                    NaN   \n",
       "A1CF                     NaN               NaN  NON_SYNONYMOUS_CODING   \n",
       "\n",
       "patient_ID HTMCP-03-06-02026        ...        HTMCP-03-06-02424  \\\n",
       "gene                                ...                            \n",
       "A1BG                     NaN        ...                      NaN   \n",
       "A1CF                     NaN        ...                      NaN   \n",
       "\n",
       "patient_ID HTMCP-03-06-02427 HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "gene                                                               \n",
       "A1BG                     NaN               NaN               NaN   \n",
       "A1CF                     NaN               NaN               NaN   \n",
       "\n",
       "patient_ID HTMCP-03-06-02435 HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "gene                                                               \n",
       "A1BG                     NaN               NaN               NaN   \n",
       "A1CF                     NaN               NaN               NaN   \n",
       "\n",
       "patient_ID HTMCP-03-06-02442 HTMCP-03-06-02447 HTMCP-03-06-02448  \n",
       "gene                                                              \n",
       "A1BG                     NaN               NaN               NaN  \n",
       "A1CF                     NaN               NaN               NaN  \n",
       "\n",
       "[2 rows x 123 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf['impact'] = rdf['impact_tmp'].apply(lambda x: impact_type(x))\n",
    "rdf = rdf[['gene', 'patient_ID', 'impact']].drop_duplicates()\n",
    "rdf = rdf.set_index(['gene', 'patient_ID'])['impact'].unstack()\n",
    "rdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "rdf = pd.concat([rdf, edf])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adenosquamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 123 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient            HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02003  \\\n",
       "HIV_status                  Positive          Negative          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02006 HTMCP-03-06-02007 HTMCP-03-06-02008  \\\n",
       "HIV_status                  Negative          Negative          Negative   \n",
       "Putative_histology          Squamous     Adenosquamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02012 HTMCP-03-06-02013 HTMCP-03-06-02020  \\\n",
       "HIV_status                  Negative          Negative          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02026        ...        HTMCP-03-06-02424  \\\n",
       "HIV_status                  Negative        ...                 Positive   \n",
       "Putative_histology             Adeno        ...                 Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02427 HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "HIV_status                  Positive          Positive          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02435 HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "HIV_status                  Positive          Positive          Positive   \n",
       "Putative_histology          Squamous          Squamous             Adeno   \n",
       "\n",
       "patient            HTMCP-03-06-02442 HTMCP-03-06-02447 HTMCP-03-06-02448  \n",
       "HIV_status                  Positive          Positive          Positive  \n",
       "Putative_histology          Squamous          Squamous             Adeno  \n",
       "\n",
       "[2 rows x 123 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['A1BG', 'A1CF', 'A2LD1', 'A2M', 'A2ML1', 'A4GALT', 'A4GNT', 'AAAS',\n",
       "       'AACS', 'AADAC',\n",
       "       ...\n",
       "       'ZXDC', 'ZYG11A', 'ZYG11B', 'ZZEF1', 'ZZZ3', 'hsa-mir-3187',\n",
       "       'hsa-mir-4763', 'yR211F11.2', 'HIV_status', 'Putative_histology'],\n",
       "      dtype='object', length=14587)"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = [ 'HIV_status', 'Putative_histology'] + genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Positive', 'Negative', 'Squamous', 'Adenosquamous', 'Adeno',\n",
       "       'Neuroendocrine', nan, 'NON_SYNONYMOUS_CODING', 'FRAME_SHIFT',\n",
       "       'Multiple', 'STOP_GAINED', 'CODON_INSERTION_DELETION'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf = rdf.reindex(genes)#.loc['PIK3CA'].value_counts()\n",
    "rdf.head(3)\n",
    "pd.Series(rdf.values.ravel()).unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "rdf['occurrence'] = (rdf.notnull().sum(axis=1)/(rdf.shape[1] -2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# genes = rdf['occurrence'].sort_values(ascending=False).index.tolist()\n",
    "# predetermined gene order"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "      <th>occurrence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>1.016529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>1.016529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLL3</th>\n",
       "      <td>CODON_INSERTION_DELETION</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.181818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLL2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.132231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CREBBP</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.082645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 124 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           HTMCP-03-06-02001 HTMCP-03-06-02002  \\\n",
       "HIV_status                          Positive          Negative   \n",
       "Putative_histology                  Squamous          Squamous   \n",
       "MLL3                CODON_INSERTION_DELETION               NaN   \n",
       "MLL2                                     NaN       STOP_GAINED   \n",
       "CREBBP                                   NaN               NaN   \n",
       "\n",
       "                   HTMCP-03-06-02003 HTMCP-03-06-02006      HTMCP-03-06-02007  \\\n",
       "HIV_status                  Positive          Negative               Negative   \n",
       "Putative_histology          Squamous          Squamous                  Adeno   \n",
       "MLL3                             NaN               NaN               Multiple   \n",
       "MLL2                             NaN               NaN                    NaN   \n",
       "CREBBP                           NaN               NaN  NON_SYNONYMOUS_CODING   \n",
       "\n",
       "                   HTMCP-03-06-02008 HTMCP-03-06-02012 HTMCP-03-06-02013  \\\n",
       "HIV_status                  Negative          Negative          Negative   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "MLL3                             NaN       STOP_GAINED               NaN   \n",
       "MLL2                             NaN               NaN               NaN   \n",
       "CREBBP                           NaN               NaN               NaN   \n",
       "\n",
       "                   HTMCP-03-06-02020 HTMCP-03-06-02026    ...      \\\n",
       "HIV_status                  Positive          Negative    ...       \n",
       "Putative_histology          Squamous             Adeno    ...       \n",
       "MLL3                     STOP_GAINED          Multiple    ...       \n",
       "MLL2                             NaN               NaN    ...       \n",
       "CREBBP                           NaN               NaN    ...       \n",
       "\n",
       "                   HTMCP-03-06-02427      HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "HIV_status                  Positive               Positive          Positive   \n",
       "Putative_histology          Squamous               Squamous          Squamous   \n",
       "MLL3                             NaN  NON_SYNONYMOUS_CODING               NaN   \n",
       "MLL2                             NaN                    NaN               NaN   \n",
       "CREBBP                           NaN                    NaN               NaN   \n",
       "\n",
       "                   HTMCP-03-06-02435 HTMCP-03-06-02437      HTMCP-03-06-02441  \\\n",
       "HIV_status                  Positive          Positive               Positive   \n",
       "Putative_histology          Squamous          Squamous                  Adeno   \n",
       "MLL3                     STOP_GAINED               NaN  NON_SYNONYMOUS_CODING   \n",
       "MLL2                     FRAME_SHIFT               NaN                    NaN   \n",
       "CREBBP                           NaN               NaN                    NaN   \n",
       "\n",
       "                   HTMCP-03-06-02442      HTMCP-03-06-02447 HTMCP-03-06-02448  \\\n",
       "HIV_status                  Positive               Positive          Positive   \n",
       "Putative_histology          Squamous               Squamous             Adeno   \n",
       "MLL3                     STOP_GAINED  NON_SYNONYMOUS_CODING               NaN   \n",
       "MLL2                             NaN                    NaN               NaN   \n",
       "CREBBP                           NaN                    NaN               NaN   \n",
       "\n",
       "                   occurrence  \n",
       "HIV_status           1.016529  \n",
       "Putative_histology   1.016529  \n",
       "MLL3                 0.181818  \n",
       "MLL2                 0.132231  \n",
       "CREBBP               0.082645  \n",
       "\n",
       "[5 rows x 124 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(rdf.loc['MLL3',:] == 'STOP_GAINED').sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# rdf = rdf.reindex(genes)\n",
    "# rdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "rdf['percentage'] = ['{0}({1}%)'.format(i[0], int(round(i[1]*100))) for i in zip(rdf.index.tolist(), rdf.occurrence)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "rdf.loc['HIV_status', 'percentage'] = 'HIV_Status'\n",
    "rdf.loc['Putative_histology', 'percentage' ] = 'Puatative_Histology'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(\"HTMCP-03-06-02061\" in rdf.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "      <th>occurrence</th>\n",
       "      <th>percentage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>1.016529</td>\n",
       "      <td>HIV_Status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>1.016529</td>\n",
       "      <td>Puatative_Histology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLL3</th>\n",
       "      <td>CODON_INSERTION_DELETION</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>...</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.181818</td>\n",
       "      <td>MLL3(18%)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           HTMCP-03-06-02001 HTMCP-03-06-02002  \\\n",
       "HIV_status                          Positive          Negative   \n",
       "Putative_histology                  Squamous          Squamous   \n",
       "MLL3                CODON_INSERTION_DELETION               NaN   \n",
       "\n",
       "                   HTMCP-03-06-02003 HTMCP-03-06-02006 HTMCP-03-06-02007  \\\n",
       "HIV_status                  Positive          Negative          Negative   \n",
       "Putative_histology          Squamous          Squamous             Adeno   \n",
       "MLL3                             NaN               NaN          Multiple   \n",
       "\n",
       "                   HTMCP-03-06-02008 HTMCP-03-06-02012 HTMCP-03-06-02013  \\\n",
       "HIV_status                  Negative          Negative          Negative   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "MLL3                             NaN       STOP_GAINED               NaN   \n",
       "\n",
       "                   HTMCP-03-06-02020 HTMCP-03-06-02026         ...           \\\n",
       "HIV_status                  Positive          Negative         ...            \n",
       "Putative_histology          Squamous             Adeno         ...            \n",
       "MLL3                     STOP_GAINED          Multiple         ...            \n",
       "\n",
       "                        HTMCP-03-06-02428 HTMCP-03-06-02434 HTMCP-03-06-02435  \\\n",
       "HIV_status                       Positive          Positive          Positive   \n",
       "Putative_histology               Squamous          Squamous          Squamous   \n",
       "MLL3                NON_SYNONYMOUS_CODING               NaN       STOP_GAINED   \n",
       "\n",
       "                   HTMCP-03-06-02437      HTMCP-03-06-02441 HTMCP-03-06-02442  \\\n",
       "HIV_status                  Positive               Positive          Positive   \n",
       "Putative_histology          Squamous                  Adeno          Squamous   \n",
       "MLL3                             NaN  NON_SYNONYMOUS_CODING       STOP_GAINED   \n",
       "\n",
       "                        HTMCP-03-06-02447 HTMCP-03-06-02448 occurrence  \\\n",
       "HIV_status                       Positive          Positive   1.016529   \n",
       "Putative_histology               Squamous             Adeno   1.016529   \n",
       "MLL3                NON_SYNONYMOUS_CODING               NaN   0.181818   \n",
       "\n",
       "                             percentage  \n",
       "HIV_status                   HIV_Status  \n",
       "Putative_histology  Puatative_Histology  \n",
       "MLL3                          MLL3(18%)  \n",
       "\n",
       "[3 rows x 125 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "sdf = rdf.drop('occurrence', axis=1).set_index('percentage', drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sdf = sdf.fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>percentage</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_Status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Puatative_Histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLL3(18%)</th>\n",
       "      <td>CODON_INSERTION_DELETION</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>...</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 123 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            HTMCP-03-06-02001 HTMCP-03-06-02002  \\\n",
       "percentage                                                        \n",
       "HIV_Status                           Positive          Negative   \n",
       "Puatative_Histology                  Squamous          Squamous   \n",
       "MLL3(18%)            CODON_INSERTION_DELETION               NaN   \n",
       "\n",
       "                    HTMCP-03-06-02003 HTMCP-03-06-02006 HTMCP-03-06-02007  \\\n",
       "percentage                                                                  \n",
       "HIV_Status                   Positive          Negative          Negative   \n",
       "Puatative_Histology          Squamous          Squamous             Adeno   \n",
       "MLL3(18%)                         NaN               NaN          Multiple   \n",
       "\n",
       "                    HTMCP-03-06-02008 HTMCP-03-06-02012 HTMCP-03-06-02013  \\\n",
       "percentage                                                                  \n",
       "HIV_Status                   Negative          Negative          Negative   \n",
       "Puatative_Histology          Squamous          Squamous          Squamous   \n",
       "MLL3(18%)                         NaN       STOP_GAINED               NaN   \n",
       "\n",
       "                    HTMCP-03-06-02020 HTMCP-03-06-02026        ...         \\\n",
       "percentage                                                     ...          \n",
       "HIV_Status                   Positive          Negative        ...          \n",
       "Puatative_Histology          Squamous             Adeno        ...          \n",
       "MLL3(18%)                 STOP_GAINED          Multiple        ...          \n",
       "\n",
       "                         HTMCP-03-06-02424 HTMCP-03-06-02427  \\\n",
       "percentage                                                     \n",
       "HIV_Status                        Positive          Positive   \n",
       "Puatative_Histology               Squamous          Squamous   \n",
       "MLL3(18%)            NON_SYNONYMOUS_CODING               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "percentage                                                     \n",
       "HIV_Status                        Positive          Positive   \n",
       "Puatative_Histology               Squamous          Squamous   \n",
       "MLL3(18%)            NON_SYNONYMOUS_CODING               NaN   \n",
       "\n",
       "                    HTMCP-03-06-02435 HTMCP-03-06-02437  \\\n",
       "percentage                                                \n",
       "HIV_Status                   Positive          Positive   \n",
       "Puatative_Histology          Squamous          Squamous   \n",
       "MLL3(18%)                 STOP_GAINED               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02441 HTMCP-03-06-02442  \\\n",
       "percentage                                                     \n",
       "HIV_Status                        Positive          Positive   \n",
       "Puatative_Histology                  Adeno          Squamous   \n",
       "MLL3(18%)            NON_SYNONYMOUS_CODING       STOP_GAINED   \n",
       "\n",
       "                         HTMCP-03-06-02447 HTMCP-03-06-02448  \n",
       "percentage                                                    \n",
       "HIV_Status                        Positive          Positive  \n",
       "Puatative_Histology               Squamous             Adeno  \n",
       "MLL3(18%)            NON_SYNONYMOUS_CODING               NaN  \n",
       "\n",
       "[3 rows x 123 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdf.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "123"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdf.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert (\"HTMCP-03-06-02061\" in sdf.columns) == False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "sdf.to_csv('/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/histone_mutations_for_oncoprint_123_patients.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
