{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import matplotlib\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.patches as patches\n",
    "from matplotlib import style, colors\n",
    "from matplotlib import gridspec\n",
    "\n",
    "matplotlib.style.use('ggplot')\n",
    "%matplotlib inline\n",
    "matplotlib.rcParams['figure.figsize'] = (8, 6)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# make oncoprint for 124 patients"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>num_patients_gene_level</th>\n",
       "      <th>num_INDELs_gene_level</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>ref_base</th>\n",
       "      <th>alt_base</th>\n",
       "      <th>num_patients_INDEL_level</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>snp_ID</th>\n",
       "      <th>...</th>\n",
       "      <th>pileup_AltC</th>\n",
       "      <th>pileup_AF</th>\n",
       "      <th>strelka_n_Cov</th>\n",
       "      <th>strelka_n_RefC</th>\n",
       "      <th>strelka_n_AltC</th>\n",
       "      <th>strelka_n_AF</th>\n",
       "      <th>strelka_t_Cov</th>\n",
       "      <th>strelka_t_RefC</th>\n",
       "      <th>strelka_t_AltC</th>\n",
       "      <th>strelka_t_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PCNT</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>47849950</td>\n",
       "      <td>CAGG</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>HTMCP-03-06-02448</td>\n",
       "      <td>novel_snp</td>\n",
       "      <td>...</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>41</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>83</td>\n",
       "      <td>62</td>\n",
       "      <td>15</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MLL3</td>\n",
       "      <td>29</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>151879476</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>HTMCP-03-06-02097</td>\n",
       "      <td>novel_snp</td>\n",
       "      <td>...</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>33</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>84</td>\n",
       "      <td>56</td>\n",
       "      <td>22</td>\n",
       "      <td>0.11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene  num_patients_gene_level  num_INDELs_gene_level chromosome   position  \\\n",
       "0  PCNT                        1                      2         21   47849950   \n",
       "1  MLL3                       29                      6          7  151879476   \n",
       "\n",
       "  ref_base alt_base  num_patients_INDEL_level         patient_ID     snp_ID  \\\n",
       "0     CAGG        C                         1  HTMCP-03-06-02448  novel_snp   \n",
       "1      CTG        C                         1  HTMCP-03-06-02097  novel_snp   \n",
       "\n",
       "      ...      pileup_AltC pileup_AF strelka_n_Cov strelka_n_RefC  \\\n",
       "0     ...               na        na            41             43   \n",
       "1     ...               na        na            33             32   \n",
       "\n",
       "  strelka_n_AltC strelka_n_AF strelka_t_Cov strelka_t_RefC strelka_t_AltC  \\\n",
       "0              0          0.0            83             62             15   \n",
       "1              0          0.0            84             56             22   \n",
       "\n",
       "  strelka_t_AF  \n",
       "0         0.05  \n",
       "1         0.11  \n",
       "\n",
       "[2 rows x 29 columns]"
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f1 ='/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/HIGH_MODERATE_INDEL_summary_with_normal_strelka_indels_only.tsv'\n",
    "indf = pd.read_csv(f1, sep='\\t')\n",
    "indf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>impact_tmp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PCNT</td>\n",
       "      <td>HTMCP-03-06-02448</td>\n",
       "      <td>CODON_DELETION</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MLL3</td>\n",
       "      <td>HTMCP-03-06-02097</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>HTMCP-03-06-02174</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>VPS13A</td>\n",
       "      <td>HTMCP-03-06-02012</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MSH6</td>\n",
       "      <td>HTMCP-03-06-02435</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene         patient_ID      impact_tmp\n",
       "0    PCNT  HTMCP-03-06-02448  CODON_DELETION\n",
       "1    MLL3  HTMCP-03-06-02097     FRAME_SHIFT\n",
       "2  MT-ND2  HTMCP-03-06-02174     FRAME_SHIFT\n",
       "3  VPS13A  HTMCP-03-06-02012     FRAME_SHIFT\n",
       "4    MSH6  HTMCP-03-06-02435     FRAME_SHIFT"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indf['impact_tmp'] = indf['snpeff_details'].apply(lambda x: x.split('(')[0])\n",
    "indf = indf[['gene', 'patient_ID','impact_tmp']]\n",
    "indf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['MLL3', 'MLL2', 'CREBBP', 'EP300', 'ATRX', 'CHD4', 'CHD3', 'ARID1A', 'HCFC1', 'EZH2', 'MLL', 'BAZ2B', 'TAF1', 'TAF1L', 'KAT6A', 'TRRAP', 'KDM5B', 'EED', 'CHD1', 'CHD2', 'SF3B1', 'CACTIN', 'CDC5L', 'CHERP', 'CRNKL1', 'DDX46', 'DHX16', 'FAM50A', 'GPKOW', 'ISY1', 'MFAP1', 'PPIL2', 'PPWD1', 'RBM25', 'SF3A3', 'SMNDC1', 'SNRPD2', 'SNW1', 'U2AF1', 'U2SURP', 'ZRSR2', 'HLA-B', 'HLA-A', 'HLA-C', 'HLA-DRB5', 'HLA-DQB2', 'HLA-DQA1', 'C6', 'HLA-DRB1', 'C1RL', 'C3', 'B2M', 'HLA-DQA2', 'FAT1', 'FAT2', 'AJUBA', 'PIK3CA', 'FBXW7', 'CASP8', 'MAPK1', 'NFE2L2', 'TP53', 'ERBB2', 'ERBB3', 'PTEN', 'STK11', 'MED1', 'ELF3', 'CBFB', 'KRAS', 'TGFBR2']\n"
     ]
    }
   ],
   "source": [
    "# get gene list\n",
    "# f2 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/genes.txt'\n",
    "f2 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/genes_grouped_for_oncoprint.txt'\n",
    "cdf = pd.read_csv(f2, comment=\"#\")\n",
    "genes = cdf.genes.tolist()\n",
    "print(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 123 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient            HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02003  \\\n",
       "HIV_status                  Positive          Negative          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02006 HTMCP-03-06-02007 HTMCP-03-06-02008  \\\n",
       "HIV_status                  Negative          Negative          Negative   \n",
       "Putative_histology          Squamous             Adeno          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02012 HTMCP-03-06-02013 HTMCP-03-06-02020  \\\n",
       "HIV_status                  Negative          Negative          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02026        ...        HTMCP-03-06-02424  \\\n",
       "HIV_status                  Negative        ...                 Positive   \n",
       "Putative_histology             Adeno        ...                 Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02427 HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "HIV_status                  Positive          Positive          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02435 HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "HIV_status                  Positive          Positive          Positive   \n",
       "Putative_histology          Squamous          Squamous             Adeno   \n",
       "\n",
       "patient            HTMCP-03-06-02442 HTMCP-03-06-02447 HTMCP-03-06-02448  \n",
       "HIV_status                  Positive          Positive          Positive  \n",
       "Putative_histology          Squamous          Squamous             Adeno  \n",
       "\n",
       "[2 rows x 123 columns]"
      ]
     },
     "execution_count": 179,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get meta track for example histology\n",
    "f3 ='/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/mutation_load_clinic.txt'\n",
    "ddf = pd.read_csv(f3, sep='\\t')\n",
    "ddf = ddf[['patient', 'HIV_status', 'Putative_histology']]\n",
    "edf = ddf.set_index('patient').T\n",
    "edf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/szong/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2717: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "# get snvs\n",
    "# f4 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/124_patients_selected_columns_new_filtered.txt'\n",
    "f4 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/HIGH_MODERATE_SNV_summary_with_normal_with_AF_filtered_somatic.tsv'\n",
    "pdf = pd.read_csv(f4, sep='\\t')\n",
    "pdf.head(2)\n",
    "pdf['impact_tmp'] = pdf['snpeff_details'].apply(lambda x: x.split('(')[0])\n",
    "mdf = pdf[['gene', 'patient_ID','impact_tmp']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>impact_tmp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>HTMCP-03-06-02214</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene         patient_ID             impact_tmp\n",
       "0  SAMD11  HTMCP-03-06-02007  NON_SYNONYMOUS_CODING\n",
       "1  SAMD11  HTMCP-03-06-02214  NON_SYNONYMOUS_CODING"
      ]
     },
     "execution_count": 181,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>impact_tmp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>HTMCP-03-06-02214</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene         patient_ID             impact_tmp\n",
       "0  SAMD11  HTMCP-03-06-02007  NON_SYNONYMOUS_CODING\n",
       "1  SAMD11  HTMCP-03-06-02214  NON_SYNONYMOUS_CODING"
      ]
     },
     "execution_count": 182,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# concat indel and snvs\n",
    "rdf = pd.concat([mdf,indf])\n",
    "rdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rdf = rdf[rdf['patient_ID']!=\"HTMCP-03-06-02061\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>patient_ID</th>\n",
       "      <th>impact_tmp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>HTMCP-03-06-02179</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>HTMCP-03-06-02020</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene         patient_ID             impact_tmp\n",
       "0  A1BG  HTMCP-03-06-02179  NON_SYNONYMOUS_CODING\n",
       "1  A1CF  HTMCP-03-06-02020  NON_SYNONYMOUS_CODING"
      ]
     },
     "execution_count": 184,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf = rdf.drop_duplicates().groupby(['gene', 'patient_ID']).agg({'impact_tmp': ','.join}).reset_index()\n",
    "rdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def impact_type(x):\n",
    "#     print(x)\n",
    "    xsplit = list(set(x.split(',')))\n",
    "    type = xsplit[0].upper()\n",
    "    if len(xsplit) > 1:\n",
    "        impact = 'Multiple'\n",
    "    elif len(xsplit) == 1:\n",
    "         if '+' in xsplit[0]:\n",
    "             impact = 'Multiple'\n",
    "         elif type == 'MISSENSE_VARIANT' or type == 'NON_SYNONYMOUS_CODING':\n",
    "             impact = 'NON_SYNONYMOUS_CODING'\n",
    "         elif  type == 'SPLICE_SITE_ACCEPTOR' or  type == 'SPLICE_SITE_DONOR':\n",
    "             impact = 'SPLICE_ACCEPTOR_DONOR'\n",
    "         elif type == 'STOP_LOST':\n",
    "             impact = 'STOP_LOST'\n",
    "         elif type == 'STOP_GAINED':\n",
    "             impact = 'STOP_GAINED'\n",
    "         elif  type == 'START_LOST':\n",
    "             impact = 'START_LOST'\n",
    "         elif type == 'START_GAINED':\n",
    "             impact = 'START_GAINED'\n",
    "         elif type == 'FRAME_SHIFT' or type == 'FRAMESHIFT_VARIANT':\n",
    "             impact = 'FRAME_SHIFT'\n",
    "         elif type == 'INFRAME_DELETION' or type == 'CODON_INSERTION' or type == 'CODON_DELETION' or type == 'CODON_CHANGE_PLUS_CODON_INSERTION' or type == 'CODON_CHANGE_PLUS_CODON_DELETION' or type.lower() == 'disruptive_inframe_deletion':\n",
    "             impact = 'CODON_INSERTION_DELETION'                \n",
    "         else: exit(1);print('ERROR')\n",
    "    return impact"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient_ID</th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 123 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient_ID HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02003  \\\n",
       "gene                                                               \n",
       "A1BG                    None              None              None   \n",
       "A1CF                    None              None              None   \n",
       "\n",
       "patient_ID HTMCP-03-06-02006 HTMCP-03-06-02007 HTMCP-03-06-02008  \\\n",
       "gene                                                               \n",
       "A1BG                    None              None              None   \n",
       "A1CF                    None              None              None   \n",
       "\n",
       "patient_ID HTMCP-03-06-02012 HTMCP-03-06-02013      HTMCP-03-06-02020  \\\n",
       "gene                                                                    \n",
       "A1BG                    None              None                   None   \n",
       "A1CF                    None              None  NON_SYNONYMOUS_CODING   \n",
       "\n",
       "patient_ID HTMCP-03-06-02026        ...        HTMCP-03-06-02424  \\\n",
       "gene                                ...                            \n",
       "A1BG                    None        ...                     None   \n",
       "A1CF                    None        ...                     None   \n",
       "\n",
       "patient_ID HTMCP-03-06-02427 HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "gene                                                               \n",
       "A1BG                    None              None              None   \n",
       "A1CF                    None              None              None   \n",
       "\n",
       "patient_ID HTMCP-03-06-02435 HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "gene                                                               \n",
       "A1BG                    None              None              None   \n",
       "A1CF                    None              None              None   \n",
       "\n",
       "patient_ID HTMCP-03-06-02442 HTMCP-03-06-02447 HTMCP-03-06-02448  \n",
       "gene                                                              \n",
       "A1BG                    None              None              None  \n",
       "A1CF                    None              None              None  \n",
       "\n",
       "[2 rows x 123 columns]"
      ]
     },
     "execution_count": 186,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf['impact'] = rdf['impact_tmp'].apply(lambda x: impact_type(x))\n",
    "rdf = rdf[['gene', 'patient_ID', 'impact']].drop_duplicates()\n",
    "rdf = rdf.set_index(['gene', 'patient_ID'])['impact'].unstack()\n",
    "rdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# of3 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/107_patients/mutations_in_oncoprint.txt'\n",
    "# rdf.fillna('NA').to_csv(of3, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rdf = pd.concat([rdf, edf])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['A1BG', 'A1CF', 'A2LD1', 'A2M', 'A2ML1', 'A4GALT', 'A4GNT', 'AAAS',\n",
       "       'AACS', 'AADAC',\n",
       "       ...\n",
       "       'ZXDC', 'ZYG11A', 'ZYG11B', 'ZZEF1', 'ZZZ3', 'hsa-mir-3187',\n",
       "       'hsa-mir-4763', 'yR211F11.2', 'HIV_status', 'Putative_histology'],\n",
       "      dtype='object', length=14587)"
      ]
     },
     "execution_count": 189,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "genes = [ 'HIV_status', 'Putative_histology'] + genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Positive', 'Negative', 'Squamous', 'Adeno',\n",
       "       'CODON_INSERTION_DELETION', None, 'Multiple', 'STOP_GAINED',\n",
       "       'NON_SYNONYMOUS_CODING', 'FRAME_SHIFT', 'SPLICE_ACCEPTOR_DONOR'], dtype=object)"
      ]
     },
     "execution_count": 191,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf = rdf.reindex(genes)#.loc['PIK3CA'].value_counts()\n",
    "rdf.head(3)\n",
    "pd.Series(rdf.values.ravel()).unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rdf['occurrence'] = (rdf.notnull().sum(axis=1)/(rdf.shape[1] -2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# genes = rdf['occurrence'].sort_values(ascending=False).index.tolist()\n",
    "# predetermined gene order"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "      <th>occurrence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>1.016529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>1.016529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLL3</th>\n",
       "      <td>CODON_INSERTION_DELETION</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>None</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>None</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>None</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>0.181818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLL2</th>\n",
       "      <td>None</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.132231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CREBBP</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.082645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 124 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           HTMCP-03-06-02001 HTMCP-03-06-02002  \\\n",
       "HIV_status                          Positive          Negative   \n",
       "Putative_histology                  Squamous          Squamous   \n",
       "MLL3                CODON_INSERTION_DELETION              None   \n",
       "MLL2                                    None       STOP_GAINED   \n",
       "CREBBP                                  None              None   \n",
       "\n",
       "                   HTMCP-03-06-02003 HTMCP-03-06-02006      HTMCP-03-06-02007  \\\n",
       "HIV_status                  Positive          Negative               Negative   \n",
       "Putative_histology          Squamous          Squamous                  Adeno   \n",
       "MLL3                            None              None               Multiple   \n",
       "MLL2                            None              None                   None   \n",
       "CREBBP                          None              None  NON_SYNONYMOUS_CODING   \n",
       "\n",
       "                   HTMCP-03-06-02008 HTMCP-03-06-02012 HTMCP-03-06-02013  \\\n",
       "HIV_status                  Negative          Negative          Negative   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "MLL3                            None       STOP_GAINED              None   \n",
       "MLL2                            None              None              None   \n",
       "CREBBP                          None              None              None   \n",
       "\n",
       "                   HTMCP-03-06-02020 HTMCP-03-06-02026    ...      \\\n",
       "HIV_status                  Positive          Negative    ...       \n",
       "Putative_histology          Squamous             Adeno    ...       \n",
       "MLL3                     STOP_GAINED          Multiple    ...       \n",
       "MLL2                            None              None    ...       \n",
       "CREBBP                          None              None    ...       \n",
       "\n",
       "                   HTMCP-03-06-02427      HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "HIV_status                  Positive               Positive          Positive   \n",
       "Putative_histology          Squamous               Squamous          Squamous   \n",
       "MLL3                            None  NON_SYNONYMOUS_CODING              None   \n",
       "MLL2                            None                   None              None   \n",
       "CREBBP                          None                   None              None   \n",
       "\n",
       "                   HTMCP-03-06-02435 HTMCP-03-06-02437      HTMCP-03-06-02441  \\\n",
       "HIV_status                  Positive          Positive               Positive   \n",
       "Putative_histology          Squamous          Squamous                  Adeno   \n",
       "MLL3                     STOP_GAINED              None  NON_SYNONYMOUS_CODING   \n",
       "MLL2                     FRAME_SHIFT              None                   None   \n",
       "CREBBP                          None              None                   None   \n",
       "\n",
       "                   HTMCP-03-06-02442      HTMCP-03-06-02447 HTMCP-03-06-02448  \\\n",
       "HIV_status                  Positive               Positive          Positive   \n",
       "Putative_histology          Squamous               Squamous             Adeno   \n",
       "MLL3                     STOP_GAINED  NON_SYNONYMOUS_CODING              None   \n",
       "MLL2                            None                   None              None   \n",
       "CREBBP                          None                   None              None   \n",
       "\n",
       "                   occurrence  \n",
       "HIV_status           1.016529  \n",
       "Putative_histology   1.016529  \n",
       "MLL3                 0.181818  \n",
       "MLL2                 0.132231  \n",
       "CREBBP               0.082645  \n",
       "\n",
       "[5 rows x 124 columns]"
      ]
     },
     "execution_count": 194,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8"
      ]
     },
     "execution_count": 209,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(rdf.loc['MLL3',:] == 'STOP_GAINED').sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [],
   "source": [
    "# rdf = rdf.reindex(genes)\n",
    "# rdf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {},
   "outputs": [],
   "source": [
    "rdf['percentage'] = ['{0}({1}%)'.format(i[0], int(round(i[1]*100))) for i in zip(rdf.index.tolist(), rdf.occurrence)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rdf.loc['HIV_status', 'percentage'] = 'HIV_Status'\n",
    "rdf.loc['Putative_histology', 'percentage' ] = 'Puatative_Histology'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 198,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(\"HTMCP-03-06-02061\" in rdf.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "      <th>occurrence</th>\n",
       "      <th>percentage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>1.016529</td>\n",
       "      <td>HIV_Status</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>1.016529</td>\n",
       "      <td>Puatative_Histology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLL3</th>\n",
       "      <td>CODON_INSERTION_DELETION</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>None</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>None</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>...</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>None</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>0.181818</td>\n",
       "      <td>MLL3(18%)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           HTMCP-03-06-02001 HTMCP-03-06-02002  \\\n",
       "HIV_status                          Positive          Negative   \n",
       "Putative_histology                  Squamous          Squamous   \n",
       "MLL3                CODON_INSERTION_DELETION              None   \n",
       "\n",
       "                   HTMCP-03-06-02003 HTMCP-03-06-02006 HTMCP-03-06-02007  \\\n",
       "HIV_status                  Positive          Negative          Negative   \n",
       "Putative_histology          Squamous          Squamous             Adeno   \n",
       "MLL3                            None              None          Multiple   \n",
       "\n",
       "                   HTMCP-03-06-02008 HTMCP-03-06-02012 HTMCP-03-06-02013  \\\n",
       "HIV_status                  Negative          Negative          Negative   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "MLL3                            None       STOP_GAINED              None   \n",
       "\n",
       "                   HTMCP-03-06-02020 HTMCP-03-06-02026         ...           \\\n",
       "HIV_status                  Positive          Negative         ...            \n",
       "Putative_histology          Squamous             Adeno         ...            \n",
       "MLL3                     STOP_GAINED          Multiple         ...            \n",
       "\n",
       "                        HTMCP-03-06-02428 HTMCP-03-06-02434 HTMCP-03-06-02435  \\\n",
       "HIV_status                       Positive          Positive          Positive   \n",
       "Putative_histology               Squamous          Squamous          Squamous   \n",
       "MLL3                NON_SYNONYMOUS_CODING              None       STOP_GAINED   \n",
       "\n",
       "                   HTMCP-03-06-02437      HTMCP-03-06-02441 HTMCP-03-06-02442  \\\n",
       "HIV_status                  Positive               Positive          Positive   \n",
       "Putative_histology          Squamous                  Adeno          Squamous   \n",
       "MLL3                            None  NON_SYNONYMOUS_CODING       STOP_GAINED   \n",
       "\n",
       "                        HTMCP-03-06-02447 HTMCP-03-06-02448 occurrence  \\\n",
       "HIV_status                       Positive          Positive   1.016529   \n",
       "Putative_histology               Squamous             Adeno   1.016529   \n",
       "MLL3                NON_SYNONYMOUS_CODING              None   0.181818   \n",
       "\n",
       "                             percentage  \n",
       "HIV_status                   HIV_Status  \n",
       "Putative_histology  Puatative_Histology  \n",
       "MLL3                          MLL3(18%)  \n",
       "\n",
       "[3 rows x 125 columns]"
      ]
     },
     "execution_count": 199,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rdf.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sdf = rdf.drop('occurrence', axis=1).set_index('percentage', drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# sdf = sdf.fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>HTMCP-03-06-02448</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>percentage</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_Status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Puatative_Histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLL3(18%)</th>\n",
       "      <td>CODON_INSERTION_DELETION</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>None</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>None</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>...</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>None</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 123 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            HTMCP-03-06-02001 HTMCP-03-06-02002  \\\n",
       "percentage                                                        \n",
       "HIV_Status                           Positive          Negative   \n",
       "Puatative_Histology                  Squamous          Squamous   \n",
       "MLL3(18%)            CODON_INSERTION_DELETION              None   \n",
       "\n",
       "                    HTMCP-03-06-02003 HTMCP-03-06-02006 HTMCP-03-06-02007  \\\n",
       "percentage                                                                  \n",
       "HIV_Status                   Positive          Negative          Negative   \n",
       "Puatative_Histology          Squamous          Squamous             Adeno   \n",
       "MLL3(18%)                        None              None          Multiple   \n",
       "\n",
       "                    HTMCP-03-06-02008 HTMCP-03-06-02012 HTMCP-03-06-02013  \\\n",
       "percentage                                                                  \n",
       "HIV_Status                   Negative          Negative          Negative   \n",
       "Puatative_Histology          Squamous          Squamous          Squamous   \n",
       "MLL3(18%)                        None       STOP_GAINED              None   \n",
       "\n",
       "                    HTMCP-03-06-02020 HTMCP-03-06-02026        ...         \\\n",
       "percentage                                                     ...          \n",
       "HIV_Status                   Positive          Negative        ...          \n",
       "Puatative_Histology          Squamous             Adeno        ...          \n",
       "MLL3(18%)                 STOP_GAINED          Multiple        ...          \n",
       "\n",
       "                         HTMCP-03-06-02424 HTMCP-03-06-02427  \\\n",
       "percentage                                                     \n",
       "HIV_Status                        Positive          Positive   \n",
       "Puatative_Histology               Squamous          Squamous   \n",
       "MLL3(18%)            NON_SYNONYMOUS_CODING              None   \n",
       "\n",
       "                         HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "percentage                                                     \n",
       "HIV_Status                        Positive          Positive   \n",
       "Puatative_Histology               Squamous          Squamous   \n",
       "MLL3(18%)            NON_SYNONYMOUS_CODING              None   \n",
       "\n",
       "                    HTMCP-03-06-02435 HTMCP-03-06-02437  \\\n",
       "percentage                                                \n",
       "HIV_Status                   Positive          Positive   \n",
       "Puatative_Histology          Squamous          Squamous   \n",
       "MLL3(18%)                 STOP_GAINED              None   \n",
       "\n",
       "                         HTMCP-03-06-02441 HTMCP-03-06-02442  \\\n",
       "percentage                                                     \n",
       "HIV_Status                        Positive          Positive   \n",
       "Puatative_Histology                  Adeno          Squamous   \n",
       "MLL3(18%)            NON_SYNONYMOUS_CODING       STOP_GAINED   \n",
       "\n",
       "                         HTMCP-03-06-02447 HTMCP-03-06-02448  \n",
       "percentage                                                    \n",
       "HIV_Status                        Positive          Positive  \n",
       "Puatative_Histology               Squamous             Adeno  \n",
       "MLL3(18%)            NON_SYNONYMOUS_CODING              None  \n",
       "\n",
       "[3 rows x 123 columns]"
      ]
     },
     "execution_count": 202,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdf.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "123"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdf.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert (\"HTMCP-03-06-02061\" in sdf.columns) == False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sdf.to_csv('/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/mutations_for_oncoprint_123_patients.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
