{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "# import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "from IPython.core.interactiveshell import InteractiveShell\n",
    "InteractiveShell.ast_node_interactivity = \"all\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "rstudio on gphost01\n",
    "http://gphost01.bcgsc.ca:8787/\n",
    "\n",
    "local rstudio to make lolliplot: use this R:\n",
    "export RSTUDIO_WHICH_R=/gsc/software/linux-x86_64-centos6/R-3.3.2/bin/R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "rstudio on gphost03\n",
    "\n",
    "http://gphost03.bcgsc.ca:8787/\n",
    "\n",
    "run specific version of R in rstudio\n",
    "\n",
    "export RSTUDIO_WHICH_R=/usr/local/bin/R\n",
    "\n",
    "\n",
    "run this command in the r script on gphost08\n",
    "/gsc/software/linux-x86_64-centos6/R-3.3.2/bin/Rscript ./rainstorm.R --input_maf /projects/trans_scratch/validations/workspace/szong/Cervical/mutsig2cv/118_patients.maf --cpu_num 8 \n",
    "\n",
    "example command to run:\n",
    "\n",
    "[szong@gphost08 mutation_rainstorm]$ /gsc/software/linux-x86_64-centos6/R-3.3.2/bin/Rscript ./rainstorm.R --input_maf /projects/trans_scratch/validations/workspace/szong/Cervical/mutsig2cv/118_patients.maf --cpu_num 48\n",
    "\n",
    "peak detection:\n",
    "[szong@gphost08 mutation_rainstorm]$ /gsc/software/linux-x86_64-centos6/R-3.3.2/bin/Rscript ./rainstorm_peaks.R --stringSplit mean_ --input_maf /projects/trans_scratch/validations/workspace/szong/Cervical/mutsig2cv/118_patients.maf  _rainstorm_k_4_mean*.tsv \n",
    "\n",
    "plots:\n",
    "/gsc/software/linux-x86_64-centos6/R-3.3.2/bin/Rscript ./rainstorm_plot.R --file_base_name mean_ _rainstorm_k_4_mean_*.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run with non-coding variants only: rainstorm has a noncoding flag. i decided to parse the maf first. this should not matter:\n",
    "\n",
    " /gsc/software/linux-x86_64-centos6/R-3.3.2/bin/Rscript ./rainstorm.R --input_maf ./118_patients_noncoding.maf --cpu_num 96 --output_base_name noncoding\n",
    " \n",
    " /gsc/software/linux-x86_64-centos6/R-3.3.2/bin/Rscript ./rainstorm_peaks.R --stringSplit mean_ --input_maf ./118_patients_noncoding.maf noncoding_rainstorm_k_4_mean*.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/projects/da_workspace/software/mutation_rainstorm/118_patients_noncoding/waveletPatientDetail_withMaf.tsv.parsed'"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wkdir = '/projects/da_workspace/software/mutation_rainstorm/118_patients_noncoding/'\n",
    "inmaf = '/projects/trans_scratch/validations/workspace/szong/Cervical/mutsig2cv/118_patients.maf'\n",
    "noncoding_maf = f'{wkdir}118_patients_noncoding.maf'\n",
    "fname = 'waveletPatientDetail_withMaf.tsv'\n",
    "infile = f'{wkdir}{fname}'\n",
    "outfile = f'{infile}.parsed'\n",
    "outfile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## pick out non-coding mutations for rainstorm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (4,40,43,71,72,83,84,85,86,87,88,90,97) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>...</th>\n",
       "      <th>ExAC_AF</th>\n",
       "      <th>ExAC_AF_AFR</th>\n",
       "      <th>ExAC_AF_AMR</th>\n",
       "      <th>ExAC_AF_EAS</th>\n",
       "      <th>ExAC_AF_FIN</th>\n",
       "      <th>ExAC_AF_NFE</th>\n",
       "      <th>ExAC_AF_OTH</th>\n",
       "      <th>ExAC_AF_SAS</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>FILTER</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4G11P</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>59734</td>\n",
       "      <td>59734</td>\n",
       "      <td>+</td>\n",
       "      <td>5'Flank</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RP11-206L10.5</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>694016</td>\n",
       "      <td>694016</td>\n",
       "      <td>+</td>\n",
       "      <td>3'Flank</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 109 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  Start_Position  \\\n",
       "0        OR4G11P               0      .     GRCh37          1           59734   \n",
       "1  RP11-206L10.5               0      .     GRCh37          1          694016   \n",
       "\n",
       "   End_Position Strand Variant_Classification Variant_Type   ...   ExAC_AF  \\\n",
       "0         59734      +                5'Flank          SNP   ...       NaN   \n",
       "1        694016      +                3'Flank          SNP   ...       NaN   \n",
       "\n",
       "  ExAC_AF_AFR ExAC_AF_AMR ExAC_AF_EAS  ExAC_AF_FIN ExAC_AF_NFE ExAC_AF_OTH  \\\n",
       "0         NaN         NaN         NaN          NaN         NaN         NaN   \n",
       "1         NaN         NaN         NaN          NaN         NaN         NaN   \n",
       "\n",
       "  ExAC_AF_SAS GENE_PHENO  FILTER  \n",
       "0         NaN        NaN    PASS  \n",
       "1         NaN        NaN    PASS  \n",
       "\n",
       "[2 rows x 109 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfmaf = pd.read_csv(inmaf, sep='\\t')\n",
    "dfmaf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2698813, 109)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfmaf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "var_classifications = [\"5'Flank\", \"3'Flank\", 'IGR', \"3'UTR\", 'Intron',\n",
    "       'Missense_Mutation', 'Silent', \"5'UTR\", 'RNA', 'Nonsense_Mutation',\n",
    "       'Splice_Site', 'Nonstop_Mutation', 'Translation_Start_Site',\n",
    "       'Frame_Shift_Ins', 'Frame_Shift_Del', 'In_Frame_Del',\n",
    "       'In_Frame_Ins']\n",
    "# coding sequence (CDS) is loosely defined. in general referrs to exons,  \n",
    "noncoding_var_classifications = [\"5'Flank\", \"3'Flank\", 'IGR', \"3'UTR\", 'Intron',\n",
    "        \"5'UTR\", 'RNA', 'Splice_Site',  'Translation_Start_Site']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_noncoding = dfmaf[dfmaf.Variant_Classification.isin(noncoding_var_classifications)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([\"5'Flank\", \"3'Flank\", 'IGR', \"3'UTR\", 'Intron', \"5'UTR\", 'RNA',\n",
       "       'Splice_Site', 'Translation_Start_Site'], dtype=object)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>...</th>\n",
       "      <th>ExAC_AF</th>\n",
       "      <th>ExAC_AF_AFR</th>\n",
       "      <th>ExAC_AF_AMR</th>\n",
       "      <th>ExAC_AF_EAS</th>\n",
       "      <th>ExAC_AF_FIN</th>\n",
       "      <th>ExAC_AF_NFE</th>\n",
       "      <th>ExAC_AF_OTH</th>\n",
       "      <th>ExAC_AF_SAS</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>FILTER</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4G11P</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>59734</td>\n",
       "      <td>59734</td>\n",
       "      <td>+</td>\n",
       "      <td>5'Flank</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RP11-206L10.5</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>694016</td>\n",
       "      <td>694016</td>\n",
       "      <td>+</td>\n",
       "      <td>3'Flank</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AL669831.1</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>737454</td>\n",
       "      <td>737454</td>\n",
       "      <td>+</td>\n",
       "      <td>3'Flank</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FAM87B</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>757477</td>\n",
       "      <td>757477</td>\n",
       "      <td>+</td>\n",
       "      <td>3'Flank</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows Ã— 109 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  Start_Position  \\\n",
       "0        OR4G11P               0      .     GRCh37          1           59734   \n",
       "1  RP11-206L10.5               0      .     GRCh37          1          694016   \n",
       "2     AL669831.1               0      .     GRCh37          1          737454   \n",
       "3         FAM87B               0      .     GRCh37          1          757477   \n",
       "\n",
       "   End_Position Strand Variant_Classification Variant_Type   ...   ExAC_AF  \\\n",
       "0         59734      +                5'Flank          SNP   ...       NaN   \n",
       "1        694016      +                3'Flank          SNP   ...       NaN   \n",
       "2        737454      +                3'Flank          SNP   ...       NaN   \n",
       "3        757477      +                3'Flank          SNP   ...       NaN   \n",
       "\n",
       "  ExAC_AF_AFR ExAC_AF_AMR ExAC_AF_EAS  ExAC_AF_FIN ExAC_AF_NFE ExAC_AF_OTH  \\\n",
       "0         NaN         NaN         NaN          NaN         NaN         NaN   \n",
       "1         NaN         NaN         NaN          NaN         NaN         NaN   \n",
       "2         NaN         NaN         NaN          NaN         NaN         NaN   \n",
       "3         NaN         NaN         NaN          NaN         NaN         NaN   \n",
       "\n",
       "  ExAC_AF_SAS GENE_PHENO  FILTER  \n",
       "0         NaN        NaN    PASS  \n",
       "1         NaN        NaN    PASS  \n",
       "2         NaN        NaN    PASS  \n",
       "3         NaN        NaN    PASS  \n",
       "\n",
       "[4 rows x 109 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_noncoding.Variant_Classification.unique()\n",
    "df_noncoding[:4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_noncoding.to_csv(noncoding_maf, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([\"5'Flank\", \"3'Flank\", 'IGR', \"3'UTR\", 'Intron',\n",
       "       'Missense_Mutation', 'Silent', \"5'UTR\", 'RNA', 'Nonsense_Mutation',\n",
       "       'Splice_Site', 'Nonstop_Mutation', 'Translation_Start_Site',\n",
       "       'Frame_Shift_Ins', 'Frame_Shift_Del', 'In_Frame_Del',\n",
       "       'In_Frame_Ins'], dtype=object)"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfmaf.Variant_Classification.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>leftPosition</th>\n",
       "      <th>rightPosition</th>\n",
       "      <th>peakPosition</th>\n",
       "      <th>peakMinValue</th>\n",
       "      <th>numberPatients</th>\n",
       "      <th>mutPerKbMaf</th>\n",
       "      <th>mostFreqVClassification</th>\n",
       "      <th>position</th>\n",
       "      <th>patient</th>\n",
       "      <th>...</th>\n",
       "      <th>ExAC_AF</th>\n",
       "      <th>ExAC_AF_AFR</th>\n",
       "      <th>ExAC_AF_AMR</th>\n",
       "      <th>ExAC_AF_EAS</th>\n",
       "      <th>ExAC_AF_FIN</th>\n",
       "      <th>ExAC_AF_NFE</th>\n",
       "      <th>ExAC_AF_OTH</th>\n",
       "      <th>ExAC_AF_SAS</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>FILTER</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1_130.1</th>\n",
       "      <td>10</td>\n",
       "      <td>118649</td>\n",
       "      <td>127947</td>\n",
       "      <td>124617</td>\n",
       "      <td>-4.725319</td>\n",
       "      <td>34</td>\n",
       "      <td>6.774922</td>\n",
       "      <td>3'Flank</td>\n",
       "      <td>118649</td>\n",
       "      <td>HTMCP.03.06.02075_T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_130.2</th>\n",
       "      <td>10</td>\n",
       "      <td>118649</td>\n",
       "      <td>127947</td>\n",
       "      <td>124617</td>\n",
       "      <td>-4.725319</td>\n",
       "      <td>34</td>\n",
       "      <td>6.774922</td>\n",
       "      <td>3'Flank</td>\n",
       "      <td>118649</td>\n",
       "      <td>HTMCP.03.06.02340_T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_130.3</th>\n",
       "      <td>10</td>\n",
       "      <td>118649</td>\n",
       "      <td>127947</td>\n",
       "      <td>124617</td>\n",
       "      <td>-4.725319</td>\n",
       "      <td>34</td>\n",
       "      <td>6.774922</td>\n",
       "      <td>3'Flank</td>\n",
       "      <td>118654</td>\n",
       "      <td>HTMCP.03.06.02075_T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_130.4</th>\n",
       "      <td>10</td>\n",
       "      <td>118649</td>\n",
       "      <td>127947</td>\n",
       "      <td>124617</td>\n",
       "      <td>-4.725319</td>\n",
       "      <td>34</td>\n",
       "      <td>6.774922</td>\n",
       "      <td>3'Flank</td>\n",
       "      <td>118680</td>\n",
       "      <td>HTMCP.03.06.02075_T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_130.5</th>\n",
       "      <td>10</td>\n",
       "      <td>118649</td>\n",
       "      <td>127947</td>\n",
       "      <td>124617</td>\n",
       "      <td>-4.725319</td>\n",
       "      <td>34</td>\n",
       "      <td>6.774922</td>\n",
       "      <td>3'Flank</td>\n",
       "      <td>118680</td>\n",
       "      <td>HTMCP.03.06.02242_T</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 124 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chromosome  leftPosition  rightPosition  peakPosition  peakMinValue  \\\n",
       "1_130.1         10        118649         127947        124617     -4.725319   \n",
       "1_130.2         10        118649         127947        124617     -4.725319   \n",
       "1_130.3         10        118649         127947        124617     -4.725319   \n",
       "1_130.4         10        118649         127947        124617     -4.725319   \n",
       "1_130.5         10        118649         127947        124617     -4.725319   \n",
       "\n",
       "         numberPatients  mutPerKbMaf mostFreqVClassification  position  \\\n",
       "1_130.1              34     6.774922                 3'Flank    118649   \n",
       "1_130.2              34     6.774922                 3'Flank    118649   \n",
       "1_130.3              34     6.774922                 3'Flank    118654   \n",
       "1_130.4              34     6.774922                 3'Flank    118680   \n",
       "1_130.5              34     6.774922                 3'Flank    118680   \n",
       "\n",
       "                     patient  ...    ExAC_AF  ExAC_AF_AFR  ExAC_AF_AMR  \\\n",
       "1_130.1  HTMCP.03.06.02075_T  ...        NaN          NaN          NaN   \n",
       "1_130.2  HTMCP.03.06.02340_T  ...        NaN          NaN          NaN   \n",
       "1_130.3  HTMCP.03.06.02075_T  ...        NaN          NaN          NaN   \n",
       "1_130.4  HTMCP.03.06.02075_T  ...        NaN          NaN          NaN   \n",
       "1_130.5  HTMCP.03.06.02242_T  ...        NaN          NaN          NaN   \n",
       "\n",
       "         ExAC_AF_EAS  ExAC_AF_FIN  ExAC_AF_NFE  ExAC_AF_OTH ExAC_AF_SAS  \\\n",
       "1_130.1          NaN          NaN          NaN          NaN         NaN   \n",
       "1_130.2          NaN          NaN          NaN          NaN         NaN   \n",
       "1_130.3          NaN          NaN          NaN          NaN         NaN   \n",
       "1_130.4          NaN          NaN          NaN          NaN         NaN   \n",
       "1_130.5          NaN          NaN          NaN          NaN         NaN   \n",
       "\n",
       "         GENE_PHENO FILTER  \n",
       "1_130.1         NaN   PASS  \n",
       "1_130.2         NaN   PASS  \n",
       "1_130.3         NaN   PASS  \n",
       "1_130.4         NaN   PASS  \n",
       "1_130.5         NaN   PASS  \n",
       "\n",
       "[5 rows x 124 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(infile, sep='\\t', low_memory=False)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df[['chromosome', ]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chromosome\n",
      "leftPosition\n",
      "rightPosition\n",
      "peakPosition\n",
      "peakMinValue\n",
      "numberPatients\n",
      "mutPerKbMaf\n",
      "mostFreqVClassification\n",
      "position\n",
      "patient\n",
      "mutrate\n",
      "unadj\n",
      "mutdiff\n",
      "SNR\n",
      "meanOfmutrate\n",
      "sdOfmutrate\n",
      "mutPerKb\n",
      "Hugo_Symbol\n",
      "Entrez_Gene_Id\n",
      "Center\n",
      "NCBI_Build\n",
      "Chromosome\n",
      "End_Position\n",
      "Strand\n",
      "Variant_Classification\n",
      "Variant_Type\n",
      "Reference_Allele\n",
      "Tumor_Seq_Allele1\n",
      "Tumor_Seq_Allele2\n",
      "dbSNP_RS\n",
      "dbSNP_Val_Status\n",
      "Matched_Norm_Sample_Barcode\n",
      "Match_Norm_Seq_Allele1\n",
      "Match_Norm_Seq_Allele2\n",
      "Tumor_Validation_Allele1\n",
      "Tumor_Validation_Allele2\n",
      "Match_Norm_Validation_Allele1\n",
      "Match_Norm_Validation_Allele2\n",
      "Verification_Status\n",
      "Validation_Status\n",
      "Mutation_Status\n",
      "Sequencing_Phase\n",
      "Sequence_Source\n",
      "Validation_Method\n",
      "Score\n",
      "BAM_File\n",
      "Sequencer\n",
      "Tumor_Sample_UUID\n",
      "Matched_Norm_Sample_UUID\n",
      "HGVSc\n",
      "HGVSp\n",
      "HGVSp_Short\n",
      "Transcript_ID\n",
      "Exon_Number\n",
      "t_depth\n",
      "t_ref_count\n",
      "t_alt_count\n",
      "n_depth\n",
      "n_ref_count\n",
      "n_alt_count\n",
      "all_effects\n",
      "Allele\n",
      "Gene\n",
      "Feature\n",
      "Feature_type\n",
      "Consequence\n",
      "cDNA_position\n",
      "CDS_position\n",
      "Protein_position\n",
      "Amino_acids\n",
      "Codons\n",
      "Existing_variation\n",
      "ALLELE_NUM\n",
      "DISTANCE\n",
      "STRAND\n",
      "SYMBOL\n",
      "SYMBOL_SOURCE\n",
      "HGNC_ID\n",
      "BIOTYPE\n",
      "CANONICAL\n",
      "CCDS\n",
      "ENSP\n",
      "SWISSPROT\n",
      "TREMBL\n",
      "UNIPARC\n",
      "RefSeq\n",
      "SIFT\n",
      "PolyPhen\n",
      "EXON\n",
      "INTRON\n",
      "DOMAINS\n",
      "GMAF\n",
      "AFR_MAF\n",
      "AMR_MAF\n",
      "ASN_MAF\n",
      "EAS_MAF\n",
      "EUR_MAF\n",
      "SAS_MAF\n",
      "AA_MAF\n",
      "EA_MAF\n",
      "CLIN_SIG\n",
      "SOMATIC\n",
      "PUBMED\n",
      "MOTIF_NAME\n",
      "MOTIF_POS\n",
      "HIGH_INF_POS\n",
      "MOTIF_SCORE_CHANGE\n",
      "IMPACT\n",
      "PICK\n",
      "VARIANT_CLASS\n",
      "TSL\n",
      "HGVS_OFFSET\n",
      "PHENO\n",
      "MINIMISED\n",
      "ExAC_AF\n",
      "ExAC_AF_AFR\n",
      "ExAC_AF_AMR\n",
      "ExAC_AF_EAS\n",
      "ExAC_AF_FIN\n",
      "ExAC_AF_NFE\n",
      "ExAC_AF_OTH\n",
      "ExAC_AF_SAS\n",
      "GENE_PHENO\n",
      "FILTER\n"
     ]
    }
   ],
   "source": [
    "for i in df.columns:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = df[['Chromosome', 'position',  'chromosome', 'leftPosition', 'rightPosition', 'mostFreqVClassification', 'patient', 'Gene', 'SYMBOL']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "df2 = df1[df1.SYMBOL=='TERT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>leftPosition</th>\n",
       "      <th>rightPosition</th>\n",
       "      <th>mostFreqVClassification</th>\n",
       "      <th>patient</th>\n",
       "      <th>Gene</th>\n",
       "      <th>SYMBOL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1_12161.1</th>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>1284668</td>\n",
       "      <td>Intron</td>\n",
       "      <td>HTMCP.03.06.02194_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_12161.2</th>\n",
       "      <td>5</td>\n",
       "      <td>1283567</td>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>1284668</td>\n",
       "      <td>Intron</td>\n",
       "      <td>HTMCP.03.06.02058_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_12161.3</th>\n",
       "      <td>5</td>\n",
       "      <td>1283582</td>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>1284668</td>\n",
       "      <td>Intron</td>\n",
       "      <td>HTMCP.03.06.02058_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_12161.4</th>\n",
       "      <td>5</td>\n",
       "      <td>1283655</td>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>1284668</td>\n",
       "      <td>Intron</td>\n",
       "      <td>HTMCP.03.06.02268_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_12161.5</th>\n",
       "      <td>5</td>\n",
       "      <td>1283705</td>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>1284668</td>\n",
       "      <td>Intron</td>\n",
       "      <td>HTMCP.03.06.02076_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chromosome  position chromosome  leftPosition  rightPosition  \\\n",
       "1_12161.1          5   1283266          5       1283266        1284668   \n",
       "1_12161.2          5   1283567          5       1283266        1284668   \n",
       "1_12161.3          5   1283582          5       1283266        1284668   \n",
       "1_12161.4          5   1283655          5       1283266        1284668   \n",
       "1_12161.5          5   1283705          5       1283266        1284668   \n",
       "\n",
       "          mostFreqVClassification              patient             Gene SYMBOL  \n",
       "1_12161.1                  Intron  HTMCP.03.06.02194_T  ENSG00000164362   TERT  \n",
       "1_12161.2                  Intron  HTMCP.03.06.02058_T  ENSG00000164362   TERT  \n",
       "1_12161.3                  Intron  HTMCP.03.06.02058_T  ENSG00000164362   TERT  \n",
       "1_12161.4                  Intron  HTMCP.03.06.02268_T  ENSG00000164362   TERT  \n",
       "1_12161.5                  Intron  HTMCP.03.06.02076_T  ENSG00000164362   TERT  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>leftPosition</th>\n",
       "      <th>rightPosition</th>\n",
       "      <th>mostFreqVClassification</th>\n",
       "      <th>patient</th>\n",
       "      <th>Gene</th>\n",
       "      <th>SYMBOL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1_1259.9</th>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>1295250</td>\n",
       "      <td>5'Flank</td>\n",
       "      <td>HTMCP.03.06.02346_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_1259.10</th>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>1295250</td>\n",
       "      <td>5'Flank</td>\n",
       "      <td>HTMCP.03.06.02377_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_1259.11</th>\n",
       "      <td>5</td>\n",
       "      <td>1295250</td>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>1295250</td>\n",
       "      <td>5'Flank</td>\n",
       "      <td>HTMCP.03.06.02148_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_1259.12</th>\n",
       "      <td>5</td>\n",
       "      <td>1295250</td>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>1295250</td>\n",
       "      <td>5'Flank</td>\n",
       "      <td>HTMCP.03.06.02176_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_1259.13</th>\n",
       "      <td>5</td>\n",
       "      <td>1295250</td>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>1295250</td>\n",
       "      <td>5'Flank</td>\n",
       "      <td>HTMCP.03.06.02259_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chromosome  position chromosome  leftPosition  rightPosition  \\\n",
       "1_1259.9           5   1295228          5       1295228        1295250   \n",
       "1_1259.10          5   1295228          5       1295228        1295250   \n",
       "1_1259.11          5   1295250          5       1295228        1295250   \n",
       "1_1259.12          5   1295250          5       1295228        1295250   \n",
       "1_1259.13          5   1295250          5       1295228        1295250   \n",
       "\n",
       "          mostFreqVClassification              patient             Gene SYMBOL  \n",
       "1_1259.9                  5'Flank  HTMCP.03.06.02346_T  ENSG00000164362   TERT  \n",
       "1_1259.10                 5'Flank  HTMCP.03.06.02377_T  ENSG00000164362   TERT  \n",
       "1_1259.11                 5'Flank  HTMCP.03.06.02148_T  ENSG00000164362   TERT  \n",
       "1_1259.12                 5'Flank  HTMCP.03.06.02176_T  ENSG00000164362   TERT  \n",
       "1_1259.13                 5'Flank  HTMCP.03.06.02259_T  ENSG00000164362   TERT  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2.head()\n",
    "df2.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "df2['new_pat'] = df2['patient'].apply(lambda x: '-'.join(x.split('_')[0].split('.')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['HTMCP-03-06-02001', 'HTMCP-03-06-02003']"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "73"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "['HTMCP-03-06-02002', 'HTMCP-03-06-02006']"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "45"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "posf = '/projects/trans_scratch/validations/workspace/szong/Cervical/73_hiv_positive_patients.txt'\n",
    "negf = '/projects/trans_scratch/validations/workspace/szong/Cervical/45_hiv_negative_patients.txt'\n",
    "pos = pd.read_csv(posf, header=None)[0].tolist()\n",
    "pos[:2]\n",
    "len(pos)\n",
    "neg = pd.read_csv(negf, header=None)[0].tolist()\n",
    "neg[:2]\n",
    "len(neg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(36, 10)"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "11"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>leftPosition</th>\n",
       "      <th>rightPosition</th>\n",
       "      <th>mostFreqVClassification</th>\n",
       "      <th>patient</th>\n",
       "      <th>Gene</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>new_pat</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1_12161.1</th>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>1284668</td>\n",
       "      <td>Intron</td>\n",
       "      <td>HTMCP.03.06.02194_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "      <td>HTMCP-03-06-02194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_12161.7</th>\n",
       "      <td>5</td>\n",
       "      <td>1283897</td>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>1284668</td>\n",
       "      <td>Intron</td>\n",
       "      <td>HTMCP.03.06.02256_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "      <td>HTMCP-03-06-02256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_12161.8</th>\n",
       "      <td>5</td>\n",
       "      <td>1283897</td>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>1284668</td>\n",
       "      <td>Intron</td>\n",
       "      <td>HTMCP.03.06.02326_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "      <td>HTMCP-03-06-02326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_12161.9</th>\n",
       "      <td>5</td>\n",
       "      <td>1284161</td>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>1284668</td>\n",
       "      <td>Intron</td>\n",
       "      <td>HTMCP.03.06.02006_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "      <td>HTMCP-03-06-02006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_12161.10</th>\n",
       "      <td>5</td>\n",
       "      <td>1284168</td>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>1284668</td>\n",
       "      <td>Intron</td>\n",
       "      <td>HTMCP.03.06.02322_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "      <td>HTMCP-03-06-02322</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_12161.19</th>\n",
       "      <td>5</td>\n",
       "      <td>1284652</td>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>1284668</td>\n",
       "      <td>Intron</td>\n",
       "      <td>HTMCP.03.06.02063_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "      <td>HTMCP-03-06-02063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_12161.23</th>\n",
       "      <td>5</td>\n",
       "      <td>1284668</td>\n",
       "      <td>5</td>\n",
       "      <td>1283266</td>\n",
       "      <td>1284668</td>\n",
       "      <td>Intron</td>\n",
       "      <td>HTMCP.03.06.02328_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "      <td>HTMCP-03-06-02328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_1259.1</th>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>1295250</td>\n",
       "      <td>5'Flank</td>\n",
       "      <td>HTMCP.03.06.02012_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "      <td>HTMCP-03-06-02012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_1259.6</th>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>1295250</td>\n",
       "      <td>5'Flank</td>\n",
       "      <td>HTMCP.03.06.02326_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "      <td>HTMCP-03-06-02326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_1259.7</th>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>1295250</td>\n",
       "      <td>5'Flank</td>\n",
       "      <td>HTMCP.03.06.02328_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "      <td>HTMCP-03-06-02328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_1259.12</th>\n",
       "      <td>5</td>\n",
       "      <td>1295250</td>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>1295250</td>\n",
       "      <td>5'Flank</td>\n",
       "      <td>HTMCP.03.06.02176_T</td>\n",
       "      <td>ENSG00000164362</td>\n",
       "      <td>TERT</td>\n",
       "      <td>HTMCP-03-06-02176</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Chromosome  position chromosome  leftPosition  rightPosition  \\\n",
       "1_12161.1           5   1283266          5       1283266        1284668   \n",
       "1_12161.7           5   1283897          5       1283266        1284668   \n",
       "1_12161.8           5   1283897          5       1283266        1284668   \n",
       "1_12161.9           5   1284161          5       1283266        1284668   \n",
       "1_12161.10          5   1284168          5       1283266        1284668   \n",
       "1_12161.19          5   1284652          5       1283266        1284668   \n",
       "1_12161.23          5   1284668          5       1283266        1284668   \n",
       "1_1259.1            5   1295228          5       1295228        1295250   \n",
       "1_1259.6            5   1295228          5       1295228        1295250   \n",
       "1_1259.7            5   1295228          5       1295228        1295250   \n",
       "1_1259.12           5   1295250          5       1295228        1295250   \n",
       "\n",
       "           mostFreqVClassification              patient             Gene  \\\n",
       "1_12161.1                   Intron  HTMCP.03.06.02194_T  ENSG00000164362   \n",
       "1_12161.7                   Intron  HTMCP.03.06.02256_T  ENSG00000164362   \n",
       "1_12161.8                   Intron  HTMCP.03.06.02326_T  ENSG00000164362   \n",
       "1_12161.9                   Intron  HTMCP.03.06.02006_T  ENSG00000164362   \n",
       "1_12161.10                  Intron  HTMCP.03.06.02322_T  ENSG00000164362   \n",
       "1_12161.19                  Intron  HTMCP.03.06.02063_T  ENSG00000164362   \n",
       "1_12161.23                  Intron  HTMCP.03.06.02328_T  ENSG00000164362   \n",
       "1_1259.1                   5'Flank  HTMCP.03.06.02012_T  ENSG00000164362   \n",
       "1_1259.6                   5'Flank  HTMCP.03.06.02326_T  ENSG00000164362   \n",
       "1_1259.7                   5'Flank  HTMCP.03.06.02328_T  ENSG00000164362   \n",
       "1_1259.12                  5'Flank  HTMCP.03.06.02176_T  ENSG00000164362   \n",
       "\n",
       "           SYMBOL            new_pat  \n",
       "1_12161.1    TERT  HTMCP-03-06-02194  \n",
       "1_12161.7    TERT  HTMCP-03-06-02256  \n",
       "1_12161.8    TERT  HTMCP-03-06-02326  \n",
       "1_12161.9    TERT  HTMCP-03-06-02006  \n",
       "1_12161.10   TERT  HTMCP-03-06-02322  \n",
       "1_12161.19   TERT  HTMCP-03-06-02063  \n",
       "1_12161.23   TERT  HTMCP-03-06-02328  \n",
       "1_1259.1     TERT  HTMCP-03-06-02012  \n",
       "1_1259.6     TERT  HTMCP-03-06-02326  \n",
       "1_1259.7     TERT  HTMCP-03-06-02328  \n",
       "1_1259.12    TERT  HTMCP-03-06-02176  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2.shape\n",
    "df2[df2.new_pat.isin(pos)&(df2.leftPosition==1283266)]['new_pat'].nunique()\n",
    "df2[df2.new_pat.isin(neg)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "df2['chr'] = 'chr'+ df2.Chromosome\n",
    "df2['end'] = df2.position + 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>position</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1_95223.1</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936082</td>\n",
       "      <td>178936083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.2</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936082</td>\n",
       "      <td>178936083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.3</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936082</td>\n",
       "      <td>178936083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.4</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936082</td>\n",
       "      <td>178936083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.5</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936082</td>\n",
       "      <td>178936083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.6</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936082</td>\n",
       "      <td>178936083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.7</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936082</td>\n",
       "      <td>178936083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.8</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936082</td>\n",
       "      <td>178936083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.9</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.10</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.11</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.12</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.13</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.14</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.15</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.16</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.17</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.18</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.19</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.20</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.21</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.22</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.23</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.24</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.25</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.26</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.27</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.28</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.29</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.30</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>178936092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.31</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936138</td>\n",
       "      <td>178936139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.32</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936180</td>\n",
       "      <td>178936181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.33</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936362</td>\n",
       "      <td>178936363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.34</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936434</td>\n",
       "      <td>178936435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.35</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936541</td>\n",
       "      <td>178936542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_95223.36</th>\n",
       "      <td>chr3</td>\n",
       "      <td>178936937</td>\n",
       "      <td>178936938</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             chr   position        end\n",
       "1_95223.1   chr3  178936082  178936083\n",
       "1_95223.2   chr3  178936082  178936083\n",
       "1_95223.3   chr3  178936082  178936083\n",
       "1_95223.4   chr3  178936082  178936083\n",
       "1_95223.5   chr3  178936082  178936083\n",
       "1_95223.6   chr3  178936082  178936083\n",
       "1_95223.7   chr3  178936082  178936083\n",
       "1_95223.8   chr3  178936082  178936083\n",
       "1_95223.9   chr3  178936091  178936092\n",
       "1_95223.10  chr3  178936091  178936092\n",
       "1_95223.11  chr3  178936091  178936092\n",
       "1_95223.12  chr3  178936091  178936092\n",
       "1_95223.13  chr3  178936091  178936092\n",
       "1_95223.14  chr3  178936091  178936092\n",
       "1_95223.15  chr3  178936091  178936092\n",
       "1_95223.16  chr3  178936091  178936092\n",
       "1_95223.17  chr3  178936091  178936092\n",
       "1_95223.18  chr3  178936091  178936092\n",
       "1_95223.19  chr3  178936091  178936092\n",
       "1_95223.20  chr3  178936091  178936092\n",
       "1_95223.21  chr3  178936091  178936092\n",
       "1_95223.22  chr3  178936091  178936092\n",
       "1_95223.23  chr3  178936091  178936092\n",
       "1_95223.24  chr3  178936091  178936092\n",
       "1_95223.25  chr3  178936091  178936092\n",
       "1_95223.26  chr3  178936091  178936092\n",
       "1_95223.27  chr3  178936091  178936092\n",
       "1_95223.28  chr3  178936091  178936092\n",
       "1_95223.29  chr3  178936091  178936092\n",
       "1_95223.30  chr3  178936091  178936092\n",
       "1_95223.31  chr3  178936138  178936139\n",
       "1_95223.32  chr3  178936180  178936181\n",
       "1_95223.33  chr3  178936362  178936363\n",
       "1_95223.34  chr3  178936434  178936435\n",
       "1_95223.35  chr3  178936541  178936542\n",
       "1_95223.36  chr3  178936937  178936938"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2[['chr', 'position', 'end']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "of = f'{wkdir}PIK3CA.bed'\n",
    "df2[['chr', 'position', 'end']].to_csv(of, index=False, sep='\\t' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>leftPosition</th>\n",
       "      <th>rightPosition</th>\n",
       "      <th>mostFreqVClassification</th>\n",
       "      <th>patient</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chromosome, leftPosition, rightPosition, mostFreqVClassification, patient]\n",
       "Index: []"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1[df1.chromosome.isnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "df1['SYMBOL'] = df1.SYMBOL.astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>SYMBOL</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chromosome</th>\n",
       "      <th>leftPosition</th>\n",
       "      <th>rightPosition</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">1</th>\n",
       "      <th>10235</th>\n",
       "      <th>17038</th>\n",
       "      <td>DDX11L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>533206</th>\n",
       "      <th>536216</th>\n",
       "      <td>RP5-857K21.3,RP5-857K21.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>725020</th>\n",
       "      <th>727962</th>\n",
       "      <td>RP11-206L10.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>755226</th>\n",
       "      <th>758168</th>\n",
       "      <td>FAM87B,LINC01128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>830098</th>\n",
       "      <th>832212</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1034108</th>\n",
       "      <th>1035272</th>\n",
       "      <td>C1orf159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1111306</th>\n",
       "      <th>1111716</th>\n",
       "      <td>TTLL10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1316309</th>\n",
       "      <th>1318167</th>\n",
       "      <td>CCNL2,RP5-890O3.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1541009</th>\n",
       "      <th>1541531</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1647901</th>\n",
       "      <th>1650472</th>\n",
       "      <td>CDK11A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2193258</th>\n",
       "      <th>2193730</th>\n",
       "      <td>SKI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2585854</th>\n",
       "      <th>2589655</th>\n",
       "      <td>TTC34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2608816</th>\n",
       "      <th>2609171</th>\n",
       "      <td>TTC34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2617146</th>\n",
       "      <th>2618218</th>\n",
       "      <td>TTC34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2627812</th>\n",
       "      <th>2631787</th>\n",
       "      <td>TTC34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2688037</th>\n",
       "      <th>2690083</th>\n",
       "      <td>TTC34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2765522</th>\n",
       "      <th>2766132</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2891195</th>\n",
       "      <th>2891214</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3097664</th>\n",
       "      <th>3098381</th>\n",
       "      <td>PRDM16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4136892</th>\n",
       "      <th>4138678</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4203107</th>\n",
       "      <th>4204726</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4391955</th>\n",
       "      <th>4392174</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4691703</th>\n",
       "      <th>4693297</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4724618</th>\n",
       "      <th>4726826</th>\n",
       "      <td>AJAP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5220251</th>\n",
       "      <th>5220378</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5727629</th>\n",
       "      <th>5735888</th>\n",
       "      <td>RP11-154H17.1,nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5875842</th>\n",
       "      <th>5877084</th>\n",
       "      <td>AL356693.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7050863</th>\n",
       "      <th>7050865</th>\n",
       "      <td>CAMTA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7342433</th>\n",
       "      <th>7344237</th>\n",
       "      <td>CAMTA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8404850</th>\n",
       "      <th>8404851</th>\n",
       "      <td>SLC45A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">X</th>\n",
       "      <th>136205070</th>\n",
       "      <th>136205818</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137527452</th>\n",
       "      <th>137528284</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138424252</th>\n",
       "      <th>138424382</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139805530</th>\n",
       "      <th>139807421</th>\n",
       "      <td>LINC00632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140877287</th>\n",
       "      <th>140878550</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140992501</th>\n",
       "      <th>140994490</th>\n",
       "      <td>MAGEC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141377240</th>\n",
       "      <th>141377776</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141637709</th>\n",
       "      <th>141638785</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141656560</th>\n",
       "      <th>141656894</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141932218</th>\n",
       "      <th>141933124</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142387895</th>\n",
       "      <th>142389670</th>\n",
       "      <td>GS1-256O22.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143930710</th>\n",
       "      <th>143932111</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144575858</th>\n",
       "      <th>144576668</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144836749</th>\n",
       "      <th>144837505</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144999423</th>\n",
       "      <th>144999581</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146132214</th>\n",
       "      <th>146133862</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146206285</th>\n",
       "      <th>146207149</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146380092</th>\n",
       "      <th>146380228</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146442089</th>\n",
       "      <th>146443984</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146643777</th>\n",
       "      <th>146644512</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147061629</th>\n",
       "      <th>147062754</th>\n",
       "      <td>FMR1NB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147276524</th>\n",
       "      <th>147277663</th>\n",
       "      <td>U40455.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147662541</th>\n",
       "      <th>147663137</th>\n",
       "      <td>AFF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149471253</th>\n",
       "      <th>149472572</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150082609</th>\n",
       "      <th>150083551</th>\n",
       "      <td>AF003626.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150240227</th>\n",
       "      <th>150241450</th>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151443147</th>\n",
       "      <th>151443934</th>\n",
       "      <td>GABRA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153013061</th>\n",
       "      <th>153014728</th>\n",
       "      <td>ABCD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154797878</th>\n",
       "      <th>154798634</th>\n",
       "      <td>TMLHE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155251445</th>\n",
       "      <th>155254978</th>\n",
       "      <td>WASH6P,WASIR1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3180 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                          SYMBOL\n",
       "chromosome leftPosition rightPosition                           \n",
       "1          10235        17038                            DDX11L1\n",
       "           533206       536216         RP5-857K21.3,RP5-857K21.2\n",
       "           725020       727962                     RP11-206L10.9\n",
       "           755226       758168                  FAM87B,LINC01128\n",
       "           830098       832212                               nan\n",
       "           1034108      1035272                         C1orf159\n",
       "           1111306      1111716                           TTLL10\n",
       "           1316309      1318167                CCNL2,RP5-890O3.3\n",
       "           1541009      1541531                              nan\n",
       "           1647901      1650472                           CDK11A\n",
       "           2193258      2193730                              SKI\n",
       "           2585854      2589655                            TTC34\n",
       "           2608816      2609171                            TTC34\n",
       "           2617146      2618218                            TTC34\n",
       "           2627812      2631787                            TTC34\n",
       "           2688037      2690083                            TTC34\n",
       "           2765522      2766132                              nan\n",
       "           2891195      2891214                              nan\n",
       "           3097664      3098381                           PRDM16\n",
       "           4136892      4138678                              nan\n",
       "           4203107      4204726                              nan\n",
       "           4391955      4392174                              nan\n",
       "           4691703      4693297                              nan\n",
       "           4724618      4726826                            AJAP1\n",
       "           5220251      5220378                              nan\n",
       "           5727629      5735888                RP11-154H17.1,nan\n",
       "           5875842      5877084                       AL356693.1\n",
       "           7050863      7050865                           CAMTA1\n",
       "           7342433      7344237                           CAMTA1\n",
       "           8404850      8404851                          SLC45A1\n",
       "...                                                          ...\n",
       "X          136205070    136205818                            nan\n",
       "           137527452    137528284                            nan\n",
       "           138424252    138424382                            nan\n",
       "           139805530    139807421                      LINC00632\n",
       "           140877287    140878550                            nan\n",
       "           140992501    140994490                         MAGEC1\n",
       "           141377240    141377776                            nan\n",
       "           141637709    141638785                            nan\n",
       "           141656560    141656894                            nan\n",
       "           141932218    141933124                            nan\n",
       "           142387895    142389670                   GS1-256O22.5\n",
       "           143930710    143932111                            nan\n",
       "           144575858    144576668                            nan\n",
       "           144836749    144837505                            nan\n",
       "           144999423    144999581                            nan\n",
       "           146132214    146133862                            nan\n",
       "           146206285    146207149                            nan\n",
       "           146380092    146380228                            nan\n",
       "           146442089    146443984                            nan\n",
       "           146643777    146644512                            nan\n",
       "           147061629    147062754                         FMR1NB\n",
       "           147276524    147277663                       U40455.1\n",
       "           147662541    147663137                           AFF2\n",
       "           149471253    149472572                            nan\n",
       "           150082609    150083551                     AF003626.1\n",
       "           150240227    150241450                            nan\n",
       "           151443147    151443934                         GABRA3\n",
       "           153013061    153014728                          ABCD1\n",
       "           154797878    154798634                          TMLHE\n",
       "           155251445    155254978                  WASH6P,WASIR1\n",
       "\n",
       "[3180 rows x 1 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.groupby(['chromosome', 'leftPosition', 'rightPosition']).agg({'SYMBOL':lambda x: ','.join(set(x))})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>leftPosition</th>\n",
       "      <th>rightPosition</th>\n",
       "      <th>patient</th>\n",
       "      <th>SYMBOL</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>nunique</th>\n",
       "      <th>&lt;lambda&gt;</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>10235</td>\n",
       "      <td>17038</td>\n",
       "      <td>37</td>\n",
       "      <td>DDX11L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>533206</td>\n",
       "      <td>536216</td>\n",
       "      <td>21</td>\n",
       "      <td>RP5-857K21.3,RP5-857K21.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome leftPosition rightPosition patient                     SYMBOL\n",
       "                                        nunique                   <lambda>\n",
       "0          1        10235         17038      37                    DDX11L1\n",
       "1          1       533206        536216      21  RP5-857K21.3,RP5-857K21.2"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>leftPosition</th>\n",
       "      <th>rightPosition</th>\n",
       "      <th>patient</th>\n",
       "      <th>SYMBOL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>10235</td>\n",
       "      <td>17038</td>\n",
       "      <td>37</td>\n",
       "      <td>DDX11L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>533206</td>\n",
       "      <td>536216</td>\n",
       "      <td>21</td>\n",
       "      <td>RP5-857K21.3,RP5-857K21.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  leftPosition  rightPosition  patient                     SYMBOL\n",
       "0          1         10235          17038       37                    DDX11L1\n",
       "1          1        533206         536216       21  RP5-857K21.3,RP5-857K21.2"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# aggregate on multiple columns\n",
    "d = {'patient': ['nunique'], 'SYMBOL':lambda x: ','.join(set(x))}\n",
    "df2 = df1.groupby(['chromosome', 'leftPosition', 'rightPosition']).agg(d).reset_index()\n",
    "df2.head(2)\n",
    "df2.columns = df2.columns.get_level_values(0)\n",
    "df2.head(2)\n",
    "df3 = df2.sort_values('patient', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "df3.to_csv(outfile, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>leftPosition</th>\n",
       "      <th>rightPosition</th>\n",
       "      <th>patient</th>\n",
       "      <th>SYMBOL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>16829943</td>\n",
       "      <td>16840992</td>\n",
       "      <td>79</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10</td>\n",
       "      <td>127574661</td>\n",
       "      <td>127589465</td>\n",
       "      <td>75</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>16</td>\n",
       "      <td>33037154</td>\n",
       "      <td>33041167</td>\n",
       "      <td>74</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>16</td>\n",
       "      <td>33063091</td>\n",
       "      <td>33066469</td>\n",
       "      <td>73</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>152073997</td>\n",
       "      <td>152087464</td>\n",
       "      <td>73</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>9</td>\n",
       "      <td>33792343</td>\n",
       "      <td>33799648</td>\n",
       "      <td>72</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5</td>\n",
       "      <td>49432029</td>\n",
       "      <td>49439927</td>\n",
       "      <td>69</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>15</td>\n",
       "      <td>22822737</td>\n",
       "      <td>22832411</td>\n",
       "      <td>66</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>3</td>\n",
       "      <td>195230629</td>\n",
       "      <td>195232990</td>\n",
       "      <td>65</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>149025692</td>\n",
       "      <td>149032010</td>\n",
       "      <td>65</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>X</td>\n",
       "      <td>88457388</td>\n",
       "      <td>88462267</td>\n",
       "      <td>65</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2</td>\n",
       "      <td>92281655</td>\n",
       "      <td>92283918</td>\n",
       "      <td>63</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>7</td>\n",
       "      <td>65053792</td>\n",
       "      <td>65060659</td>\n",
       "      <td>63</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>4</td>\n",
       "      <td>3557457</td>\n",
       "      <td>3571431</td>\n",
       "      <td>62</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15</td>\n",
       "      <td>20136470</td>\n",
       "      <td>20138745</td>\n",
       "      <td>62</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4</td>\n",
       "      <td>49153176</td>\n",
       "      <td>49154983</td>\n",
       "      <td>62</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>9</td>\n",
       "      <td>45357066</td>\n",
       "      <td>45366189</td>\n",
       "      <td>61</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>10</td>\n",
       "      <td>42638043</td>\n",
       "      <td>42640403</td>\n",
       "      <td>60</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>X</td>\n",
       "      <td>155239</td>\n",
       "      <td>162703</td>\n",
       "      <td>60</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>7</td>\n",
       "      <td>142458277</td>\n",
       "      <td>142469657</td>\n",
       "      <td>59</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>7</td>\n",
       "      <td>64960477</td>\n",
       "      <td>64962979</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>16</td>\n",
       "      <td>33862919</td>\n",
       "      <td>33863718</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>3</td>\n",
       "      <td>195668321</td>\n",
       "      <td>195672230</td>\n",
       "      <td>58</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>15</td>\n",
       "      <td>20540635</td>\n",
       "      <td>20549271</td>\n",
       "      <td>58</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>16</td>\n",
       "      <td>32834165</td>\n",
       "      <td>32836529</td>\n",
       "      <td>55</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>3</td>\n",
       "      <td>195221220</td>\n",
       "      <td>195222797</td>\n",
       "      <td>55</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>2</td>\n",
       "      <td>87642518</td>\n",
       "      <td>87647158</td>\n",
       "      <td>55</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>2</td>\n",
       "      <td>92241048</td>\n",
       "      <td>92242294</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>3</td>\n",
       "      <td>197110988</td>\n",
       "      <td>197118377</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>X</td>\n",
       "      <td>9371453</td>\n",
       "      <td>9382440</td>\n",
       "      <td>54</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome  leftPosition  rightPosition  patient  SYMBOL\n",
       "0           1      16829943       16840992       79       3\n",
       "1          10     127574661      127589465       75       3\n",
       "2          16      33037154       33041167       74       1\n",
       "3          16      33063091       33066469       73       1\n",
       "4           7     152073997      152087464       73       1\n",
       "5           9      33792343       33799648       72       1\n",
       "6           5      49432029       49439927       69       0\n",
       "7          15      22822737       22832411       66       1\n",
       "8           3     195230629      195232990       65       0\n",
       "9           1     149025692      149032010       65       1\n",
       "10          X      88457388       88462267       65       0\n",
       "11          2      92281655       92283918       63       0\n",
       "12          7      65053792       65060659       63       0\n",
       "13          4       3557457        3571431       62       0\n",
       "14         15      20136470       20138745       62       0\n",
       "15          4      49153176       49154983       62       0\n",
       "16          9      45357066       45366189       61       1\n",
       "17         10      42638043       42640403       60       1\n",
       "18          X        155239         162703       60       0\n",
       "19          7     142458277      142469657       59       2\n",
       "20          7      64960477       64962979       58       0\n",
       "21         16      33862919       33863718       58       0\n",
       "22          3     195668321      195672230       58       2\n",
       "23         15      20540635       20549271       58       1\n",
       "24         16      32834165       32836529       55       1\n",
       "25          3     195221220      195222797       55       0\n",
       "26          2      87642518       87647158       55       0\n",
       "27          2      92241048       92242294       54       0\n",
       "28          3     197110988      197118377       54       0\n",
       "29          X       9371453        9382440       54       1"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.groupby(['chromosome', 'leftPosition', 'rightPosition'])['patient', 'SYMBOL'].nunique().sort_values('patient', ascending=False).reset_index().head(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "chromosome  leftPosition  rightPosition\n",
       "1           16829943      16840992         79\n",
       "10          127574661     127589465        75\n",
       "16          33037154      33041167         74\n",
       "            33063091      33066469         73\n",
       "7           152073997     152087464        73\n",
       "9           33792343      33799648         72\n",
       "5           49432029      49439927         69\n",
       "15          22822737      22832411         66\n",
       "1           149025692     149032010        65\n",
       "X           88457388      88462267         65\n",
       "3           195230629     195232990        65\n",
       "2           92281655      92283918         63\n",
       "7           65053792      65060659         63\n",
       "4           49153176      49154983         62\n",
       "15          20136470      20138745         62\n",
       "4           3557457       3571431          62\n",
       "9           45357066      45366189         61\n",
       "10          42638043      42640403         60\n",
       "X           155239        162703           60\n",
       "7           142458277     142469657        59\n",
       "            64960477      64962979         58\n",
       "3           195668321     195672230        58\n",
       "15          20540635      20549271         58\n",
       "16          33862919      33863718         58\n",
       "            32834165      32836529         55\n",
       "2           87642518      87647158         55\n",
       "3           195221220     195222797        55\n",
       "2           92241048      92242294         54\n",
       "X           9371453       9382440          54\n",
       "11          55012314      55018381         54\n",
       "Name: patient, dtype: int64"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.groupby(['chromosome', 'leftPosition', 'rightPosition'])['patient'].nunique().sort_values(ascending=False).head(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
